---
title: "Analysis of TCR repertoires of LckWT, LckCA and LckCA/KR mice"
author: "Veronika Niederlova"
date: "8/23/2022"
output: rmdformats::material
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(include = TRUE, warning = FALSE, message = FALSE, error = TRUE, cache = TRUE)

library(tidyverse)
library(readr)
library(readxl)
library(cowplot)
library(patchwork)
library(immunarch)
library(pheatmap)
library(factoextra)
library(kableExtra)
library(SmartEDA)

ggtheme <- function() {
  theme(
    axis.text = element_text(size = 10),
    axis.title = element_text(size = 10),
    text = element_text(size = 10, colour = "black"),
    legend.text = element_text(size = 10),
    legend.key.size =  unit(10, units = "points"),
    axis.text.x = element_text(angle = 90)) 
  }

binary <- function(x){
  x <- ifelse(x>0,1,0)
  return(x)
}

```


# Initial processing

This document describes the analysis of TCR repertoires for the manuscript *"Unique roles of coreceptor-bound LCK in helper and cytotoxic T cells"* by Horkova et al. The analysis workflow included: 

* sorting of CD4 and CD8 cells from lymph nodes and thymi of *Lck^WT^*, *Lck^CA^* and *Lck^CA/KR^* mice,
* isolation of RNA using [RNA Clean & Concentrator-5 kit](https://www.integratedsci.com.au/product/rna-clean-and-concentrator-5-kit.html)
* preparation of TCR-enriched libraries using the [NEBNextÂ® Mouse Immune Sequencing Kit](https://international.neb.com/products/e6330-nebnext-immune-sequencing-kit-mouse#Product%20Information_Reagents-Supplied)
* sequencing the libraries on Illumina Miseq, 300bp paired-end reads
* demultiplexing reads in Illumina BaseSpace 
* processing fastq files using the recommended [pRESTO workflow](https://international.neb.com/-/media/nebus/files/nebnext-chart-pdfs/nebnext-immune-sequencing-data-analysis-workflow-mouse.pdf?rev=7f21b23a3ce444e3bc4dcd57282402ba&hash=C0AE101CBA6C31B2A897A207D9EA6A4D) on Galaxy with default parameters
* aligning processed fastq files using [MiXCR](https://mixcr.readthedocs.io/en/develop/)

Raw data are deposited in the [Sequence Read Archive (PRJNA872031)](https://www.ncbi.nlm.nih.gov/bioproject/PRJNA872031). Processed files can be downloaded [from Zenodo](https://zenodo.org/record/7188457). The Zenodo archive contains the following files, which are needed to run this script:

* merged outputs from MiXCR `merged_TCR_repertoires.csv`
* metadata file `metadata_Lck.csv`
* TRA and TRB repertoires prepared for processing with the Immunarch package `immdata_tra.rds`, `immdata_trb.rds`

Let's read the required files:

```{r}

merged_repertoires <- read.csv("merged_TCR_repertoires.csv")
md <- read.csv("metadata_Lck.csv")

```

# Count tables

We will start with processing of the file `merged_TCR_repertoires.csv`. This file contains outputs of MiXCR software, i.e., all the assembled CDR3 clonotypes, their counts in each sample, nucleotide and amino-acid sequences, all V, D and J alignments with scores, together with some meta-data. the column `num_id` contains info about the sample number, which we will frequently use to attach meta-data to sample numbers.

Here, we summarize the variables in the file `merged_TCR_repertoires.csv`:

```{r}
kable(ExpData(data=merged_repertoires, type=2)) %>%
  kable_styling(full_width = F, font_size = 11, 
                bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```

First, we will generate summary count tables for TRA and TRB CDR3 sequences (these tables are provided as Supplementary table S1 in the manuscript). We will reorder the file so that there will be one column for each sample, and we will change the `num_id` of the samples to meaningful names. 

```{r}
count_table_tra <- merged_repertoires %>% 
  dplyr::filter(chain == "TRA") %>% 
  dplyr::select(aaSeqCDR3, num_id, cloneCount)

counts_tra <- count_table_tra %>% 
  pivot_wider(names_from = num_id, values_from = cloneCount, values_fill = 0, values_fn = sum)

md2_tra <- merged_repertoires %>% filter(chain == "TRA") %>% group_by(aaSeqCDR3) %>% slice_head(n = 1)

excel_count_table_tra <- counts_tra %>% left_join(md2_tra %>% select(allVHitsWithScore, allDHitsWithScore, allJHitsWithScore))

md_to_join <- md %>% mutate(new_name = paste(Cell_type, Organ, Mouse_strain, paste0("Exp0",Exp)))  %>% select(num_id, new_name) %>% arrange(num_id)

order <- match(as.numeric(colnames(counts_tra)[2:29]), md_to_join$num_id)

colnames(excel_count_table_tra)[2:29] <- pull(md_to_join, new_name)[order]
excel_count_table_tra2 <- excel_count_table_tra %>% dplyr::select(1, 31, 32, 30, 22, 12, 25, 11, 6, 26, 13, 18, 4, 28, 17, 14, 8, 23, 9, 7, 16, 24, 2, 21, 5, 29, 10, 20, 19, 3, 15, 27)
```

Here, we show the resulting table for TRA:

```{r}
kable(ExpData(data=excel_count_table_tra2, type=2)) %>%
  kable_styling(full_width = F, font_size = 11, 
                bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```


```{r}
count_table_trb <- merged_repertoires %>% 
  dplyr::filter(chain == "TRB") %>% 
  dplyr::select(aaSeqCDR3, num_id, cloneCount)

counts_trb <- count_table_trb %>% 
  pivot_wider(names_from = num_id, values_from = cloneCount, values_fill = 0, values_fn = sum)

md2_trb <- merged_repertoires %>% 
  filter(chain == "TRB") %>% 
  group_by(aaSeqCDR3) %>% 
  slice_head(n = 1)

excel_count_table_trb <- counts_trb %>% 
  left_join(md2_trb %>% 
              select(allVHitsWithScore, allDHitsWithScore, allJHitsWithScore))

md_to_join <- md %>% 
  mutate(new_name = paste(Cell_type, Organ, Mouse_strain, paste0("Exp0",Exp))) %>% 
  select(num_id, new_name) %>% 
  arrange(num_id)

order <- match(as.numeric(colnames(counts_trb)[2:29]), md_to_join$num_id)

colnames(excel_count_table_trb)[2:29] <- pull(md_to_join, new_name)[order]

excel_count_table_trb2 <- excel_count_table_trb %>% 
  dplyr::select(1, 30:32, 22, 12, 25, 11, 6, 26, 13, 18, 4, 28, 17, 14, 8, 23, 9, 7, 16, 24, 2, 21, 5, 29, 10, 20, 19, 3, 15, 27)
```

Here, we show the resulting table for TRB:

```{r}
kable(ExpData(data=excel_count_table_trb2, type=2)) %>%
  kable_styling(full_width = F, font_size = 11, 
                bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```


In addition to count matrices, we will also need normalized frequency tables, in which we will show the percentage of the repertoire occupied by each CDR3 sequence in a particular sample. We will first just normalize the count matrices we created earlier. 

For TRA, we will also remove canonical NKT-cell CDR3 sequences TRAV11-TRAJ18 before normalization. 

Here comes the normalized table for TRA:

```{r}

excel_count_table_tra3 <- excel_count_table_tra2 %>% 
  mutate(nkt_trav11_traj18 = if_else(
    (grepl(allVHitsWithScore, pattern = "TRAV11") & 
       (grepl(allJHitsWithScore, pattern = "TRAJ18"))),"yes","no"))

excel_count_table_tra4 <- excel_count_table_tra3 %>% 
  filter(nkt_trav11_traj18 == "no")

count_table_tra4 <- as.matrix(excel_count_table_tra4[,5:32])
rownames(count_table_tra4) <- excel_count_table_tra4$aaSeqCDR3

tra4_norm <- scale(count_table_tra4, center=FALSE, scale=colSums(count_table_tra4))

prop.table.tra <- cbind(tra4_norm, 
                        excel_count_table_tra4 %>% 
                          select(-starts_with("CD")) ) 

prop.table.tra2 <- prop.table.tra %>% 
  dplyr::select(29, 31, 30, 32, 1:28)

data.table(prop.table.tra2 %>% head) 
```

Here comes the normalized table for TRA:

```{r}
excel_count_table_trb3 <- excel_count_table_trb2 %>% 
  mutate(nkt_trav11_traj18 = "no")

count_table_trb4 <- as.matrix(excel_count_table_trb3[,5:32])
rownames(count_table_trb4) <- excel_count_table_trb3$aaSeqCDR3


trb4_norm <- scale(count_table_trb4, center=FALSE, scale=colSums(count_table_trb4))

prop.table.trb <- cbind(trb4_norm, excel_count_table_trb %>% select(-starts_with("CD")) ) 
colnames(prop.table.trb)
prop.table.trb2 <- prop.table.trb %>% dplyr::select(29, 31, 30, 32, 1:28)


data.table(prop.table.trb2 %>% head)
```

# Gene segment usage analysis

To compare the gene segments used by T-cells from different mouse strains, we will use the R package `Immunarch`. First, we loaded the MiXCR outputs using the `repLoad` function and saved it as `immdata_tra` and `immdata_trb`, respectively to facilitate the analysis. We will load the prepared datasets:

```{r}
immdata_tra <- readRDS("immdata_tra.rds")
immdata_trb <- readRDS("immdata_trb.rds")
```

Using the gene usage visualization from the `Immunarch` package, we will plot the gene segments of TRA chains and TRB chains for each cell type and each organ. 

### TRA CD4 Thymus

```{r fig.width=25, fig.height=8}
### Thymus

cd4_thymus <- geneUsage(repFilter(immdata_tra, 
                                 .method = "by.meta", 
                                 .query = list(Organ = include('Thymus'), Cell_type = include('CD4')))$data, .norm = T, .ambig = "maj", .quant = "count", .gene = "musmus.trav")

vis(cd4_thymus, .by = c("Cell_type", "Organ","Mouse_strain"), .meta = immdata_tra$meta) + ggtitle("TRA Thymus CD4") + scale_fill_manual(values = c("dodgerblue","indianred2","gray70"))

# source data
write.csv(cd4_thymus, "source_data/FigED7_cd4_thymus_tra.csv")


#ggsave("./final_fig/gene_usage_new/cd4_thymus_tra_just1segment.png", width = 60, height = 12, units = "cm") 
#ggsave("./final_fig/gene_usage_new/cd4_thymus_tra_just1segment.svg", width = 60, height = 12, units = "cm")
```

### TRA CD4 Lymph nodes

```{r fig.width=25, fig.height=8}
### LN
cd4_ln <- geneUsage(repFilter(immdata_tra, 
                                 .method = "by.meta", 
                                 .query = list(Organ = exclude('Thymus'), Cell_type = include('CD4')))$data, .norm = T, .ambig = "maj", .quant = "count", .gene = "musmus.trav")

vis(cd4_ln, .by = c("Cell_type", "Organ","Mouse_strain"), .meta = immdata_tra$meta) + ggtitle("TRA LN CD4") + scale_fill_manual(values = c("dodgerblue","indianred2","gray70"))

# source data
write.csv(cd4_ln, "source_data/FigED7_cd4_ln_tra.csv")

#ggsave("./final_fig/gene_usage_new/cd4_ln_tra_just1segment.png", width = 60, height = 12, units = "cm")
#ggsave("./final_fig/gene_usage_new/cd4_ln_tra_just1segment.svg", width = 60, height = 12, units = "cm")
```

### TRA CD8 Thymus

```{r fig.width=25, fig.height=8}
cd8_thymus <- geneUsage(repFilter(immdata_tra, 
                                 .method = "by.meta", 
                                 .query = list(Organ = include('Thymus'), Cell_type = include('CD8')))$data, .norm = T, .ambig = "maj", .quant = "count", .gene = "musmus.trav")

vis(cd8_thymus, .by = c("Cell_type", "Organ","Mouse_strain"), .meta = immdata_tra$meta) + ggtitle("TRA Thymus CD8") + scale_fill_manual(values = c("dodgerblue","indianred2","gray70"))

# source data
write.csv(cd8_thymus, "source_data/FigED7_cd8_thymus_tra.csv")

#ggsave("./final_fig/gene_usage_new/cd8_thymus_tra_just1segment.png", width = 60, height = 12, units = "cm")
#ggsave("./final_fig/gene_usage_new/cd8_thymus_tra_just1segment.svg", width = 60, height = 12, units = "cm")
```

### TRA CD8 Lymph nodes

```{r fig.width=25, fig.height=8}
cd8_ln <- geneUsage(repFilter(immdata_tra, 
                                 .method = "by.meta", 
                                 .query = list(Organ = exclude('Thymus'), Cell_type = include('CD8')))$data, .norm = T, .ambig = "maj", .quant = "count", .gene = "musmus.trav")

vis(cd8_ln, .by = c("Cell_type", "Organ","Mouse_strain"), .meta = immdata_tra$meta) + ggtitle("TRA LN CD8") + scale_fill_manual(values = c("dodgerblue","indianred2","gray70"))

# source data
write.csv(cd8_ln, "source_data/FigED7_cd8_ln_tra.csv")

#ggsave("./final_fig/gene_usage_new/cd8_ln_tra_just1segment.png", width = 60, height = 12, units = "cm")
#ggsave("./final_fig/gene_usage_new/cd8_ln_tra_just1segment.svg", width = 60, height = 12, units = "cm")


```

### TRB CD4 Thymus

```{r fig.width=8, fig.height=4}
### Thymus

cd4_thymus <- geneUsage(repFilter(immdata_trb, 
                                 .method = "by.meta", 
                                 .query = list(Organ = include('Thymus'), Cell_type = include('CD4')))$data, .norm = T, .ambig = "maj", .quant = "count", .gene = "musmus.trbv")

vis(cd4_thymus, .by = c("Cell_type", "Organ","Mouse_strain"), .meta = immdata_trb$meta, .test = F) + ggtitle("TRB Thymus CD4") + scale_fill_manual(values = c("dodgerblue","indianred2","gray70")) 
  
# source data
write.csv(cd4_thymus, "source_data/FigED7_cd4_thymus_trb.csv")

#ggsave("./final_fig/gene_usage_new/cd4_thymus_trb_just1segment.png", width = 20, height = 10, units = "cm") 
#ggsave("./final_fig/gene_usage_new/cd4_thymus_trb_just1segment.svg", width = 20, height = 10, units = "cm")
```

### TRB CD4 Lymph nodes

```{r fig.width=8, fig.height=4}
### LN
cd4_ln <- geneUsage(repFilter(immdata_trb, 
                                 .method = "by.meta", 
                                 .query = list(Organ = exclude('Thymus'), Cell_type = include('CD4')))$data, .norm = T, .ambig = "maj", .quant = "count", .gene = "musmus.trbv")

vis(cd4_ln, .by = c("Cell_type", "Organ","Mouse_strain"), .meta = immdata_trb$meta, .test = F) + ggtitle("TRB LN CD4") + scale_fill_manual(values = c("dodgerblue","indianred2","gray70"))

# source data
write.csv(cd4_ln, "source_data/FigED7_cd4_ln_trb.csv")

#ggsave("./final_fig/gene_usage_new/cd4_ln_trb_just1segment.png", width = 20, height = 10, units = "cm")
#ggsave("./final_fig/gene_usage_new/cd4_ln_trb_just1segment.svg", width = 20, height = 10, units = "cm")
```

### TRB CD8 Thymus

```{r fig.width=8, fig.height=4}
cd8_thymus <- geneUsage(repFilter(immdata_trb, 
                                 .method = "by.meta", 
                                 .query = list(Organ = include('Thymus'), Cell_type = include('CD8')))$data, .norm = T, .ambig = "maj", .quant = "count", .gene = "musmus.trbv")

vis(cd8_thymus, .by = c("Cell_type", "Organ","Mouse_strain"), .meta = immdata_trb$meta, .test = F) + ggtitle("TRB Thymus CD8") + scale_fill_manual(values = c("dodgerblue","indianred2","gray70"))

# source data
write.csv(cd8_thymus, "source_data/FigED7_cd8_thymus_trb.csv")

#ggsave("./final_fig/gene_usage_new/cd8_thymus_trb_just1segment.png", width = 20, height = 10, units = "cm")
#ggsave("./final_fig/gene_usage_new/cd8_thymus_trb_just1segment.svg", width = 20, height = 10, units = "cm")
```

### TRB CD8 Lymph nodes

```{r fig.width=8, fig.height=4}
cd8_ln <- geneUsage(repFilter(immdata_trb, 
                                 .method = "by.meta", 
                                 .query = list(Organ = exclude('Thymus'), Cell_type = include('CD8')))$data, .norm = T, .ambig = "maj", .quant = "count", .gene = "musmus.trbv")

vis(cd8_ln, .by = c("Cell_type", "Organ","Mouse_strain"), .meta = immdata_trb$meta, .test = F) + ggtitle("TRB LN CD8") + scale_fill_manual(values = c("dodgerblue","indianred2","gray70"))

# source data
write.csv(cd8_ln, "source_data/FigED7_cd8_ln_trb.csv")

#ggsave("./final_fig/gene_usage_new/cd8_ln_trb_just1segment.png", width = 20, height = 10, units = "cm")
#ggsave("./final_fig/gene_usage_new/cd8_ln_trb_just1segment.svg", width = 20, height = 10, units = "cm")

```

# Analysis of NKT cells

Next, we will identify gene segments that are typical for invariant NKT cells. These segments are TRAV11 and TRAJ18 for TRA and TRBV1, TRBV13-2 and TRBV29 for TRB. Please, note that some of the included TRB chains are used by conventional cells as well. 

```{r}
# NKT analysis
merged_repertoires <- merged_repertoires %>% 
  mutate(is_nkt = if_else(
    (grepl(allVHitsWithScore, pattern = "TRAV11")) & (grepl(allJHitsWithScore, pattern = "TRAJ18")) |
    (grepl(allVHitsWithScore, pattern = "TRBV13-2")) |
      (grepl(allVHitsWithScore, pattern = "TRBV1\\*")) |
      (grepl(allVHitsWithScore, pattern = "TRBV29"))  ,"yes","no")) %>% 
  mutate(nkt_trav11_traj18 = if_else(
    (grepl(allVHitsWithScore, pattern = "TRAV11") & (grepl(allJHitsWithScore, pattern = "TRAJ18"))),"yes","no")) %>% mutate(nkt_trbv13_2 = if_else(
    (grepl(allVHitsWithScore, pattern = "TRBV13-2")),"yes","no")) %>% mutate(nkt_trbv29 = if_else(
    (grepl(allVHitsWithScore, pattern = "TRBV29")),"yes","no")) %>% mutate(nkt_trbv1 = if_else(
    (grepl(allVHitsWithScore, pattern = "TRBV1\\*")),"yes","no"))

md <- md %>% select(Exp, Organ, Cell_type, Mouse_strain, num_id) %>% mutate_at(vars("num_id"), as.numeric)

# All TRB NKT
nkt_table_trb <- merged_repertoires %>% 
   filter(chain == "TRB") %>% 
 mutate(new_name = paste(Exp, Organ, Cell_type, Mouse_strain)) %>% 
  dplyr::select(new_name, cloneCount, is_nkt, num_id) %>% 
  uncount(cloneCount) %>% 
  group_by(num_id, is_nkt) %>% 
    summarise(n = n()) %>% 
mutate(freq = n / sum(n)) %>% dplyr::filter(is_nkt == "yes") %>% 
  ungroup %>% 
  left_join(md) %>% 
  unique %>%
   mutate(sample_type = paste(Cell_type, Organ, Mouse_strain)) 

# TRAV11 TRAJ18
nkt_table_trav11_traj18 <- merged_repertoires %>% 
   filter(chain == "TRA") %>% 
 mutate(new_name = paste(Exp, Organ, Cell_type, Mouse_strain)) %>% 
  dplyr::select(new_name, cloneCount, nkt_trav11_traj18, num_id) %>% 
  uncount(cloneCount) %>% 
  group_by(num_id, nkt_trav11_traj18) %>% 
    summarise(n = n()) %>% 
mutate(freq = n / sum(n)) %>% dplyr::filter(nkt_trav11_traj18 == "yes") %>% 
  ungroup %>% 
  left_join(md) %>% 
  unique %>%
   mutate(sample_type = paste(Cell_type, Organ, Mouse_strain)) 

# TRBV1
nkt_table_trbv1 <- merged_repertoires %>% 
   filter(chain == "TRB") %>% 
 mutate(new_name = paste(Exp, Organ, Cell_type, Mouse_strain)) %>% 
  dplyr::select(new_name, cloneCount, nkt_trbv1, num_id) %>% 
  uncount(cloneCount) %>% 
  group_by(num_id, nkt_trbv1) %>% 
    summarise(n = n()) %>% 
mutate(freq = n / sum(n)) %>% dplyr::filter(nkt_trbv1 == "yes") %>% 
  ungroup %>% 
  left_join(md) %>% 
  unique %>%
   mutate(sample_type = paste(Cell_type, Organ, Mouse_strain)) 

# TRBV13-2
nkt_table_trbv13_2 <- merged_repertoires %>% 
   filter(chain == "TRB") %>% 
 mutate(new_name = paste(Exp, Organ, Cell_type, Mouse_strain)) %>% 
  dplyr::select(new_name, cloneCount, nkt_trbv13_2, num_id) %>% 
  uncount(cloneCount) %>% 
  group_by(num_id, nkt_trbv13_2) %>% 
    summarise(n = n()) %>% 
mutate(freq = n / sum(n)) %>% dplyr::filter(nkt_trbv13_2 == "yes") %>% 
  ungroup %>% 
  left_join(md) %>% 
  unique %>%
   mutate(sample_type = paste(Cell_type, Organ, Mouse_strain)) 


# TRBV29
nkt_table_trbv29 <- merged_repertoires %>% 
   filter(chain == "TRB") %>% 
 mutate(new_name = paste(Exp, Organ, Cell_type, Mouse_strain)) %>% 
  dplyr::select(new_name, cloneCount, nkt_trbv29, num_id) %>% 
  uncount(cloneCount) %>% 
  group_by(num_id, nkt_trbv29) %>% 
    summarise(n = n()) %>% 
mutate(freq = n / sum(n)) %>% dplyr::filter(nkt_trbv29 == "yes") %>% 
  ungroup %>% 
  left_join(md) %>% 
  unique %>%
   mutate(sample_type = paste(Cell_type, Organ, Mouse_strain)) 

levels_cd4 <- c("CD4 Thymus WT", "CD4 Thymus CA", "CD4 Thymus CAKR", "CD4 Lymph nodes WT", "CD4 Lymph nodes CA", "CD4 Lymph nodes CAKR")
levels_cd8 <- c("CD8 Thymus WT", "CD8 Thymus CA", "CD8 Thymus CAKR", "CD8 Lymph nodes WT", "CD8 Lymph nodes CA", "CD8 Lymph nodes CAKR")

nkt_table_trb$nkt_sample <- "all_nkt_trb"
nkt_table_trav11_traj18$nkt_sample <- "trav11_traj18"
nkt_table_trbv1$nkt_sample <- "trbv1"
nkt_table_trbv13_2$nkt_sample <- "trbv13_2"
nkt_table_trbv29$nkt_sample <- "trbv29"

nkt_table <- cbind(nkt_table_trb$num_id, nkt_table_trb$sample_type, nkt_table_trb$Exp, format(nkt_table_trb$freq*100, digits = 2), 
                   format(nkt_table_trav11_traj18$freq*100, digits = 2),
format(nkt_table_trbv1$freq*100, digits = 2), 
format(nkt_table_trbv13_2$freq*100, digits = 2),
                   format(nkt_table_trbv29$freq*100, digits = 2)) %>% as.data.frame

colnames(nkt_table) <- c("num_id","Sample","Exp","All TRB NKT", "TRAV11-TRAJ18", "TRBV1","TRBV13-2","TRBV29")
nkt_table <- nkt_table %>% mutate(Sample = paste(Sample, Exp)) %>% select(-num_id, -Exp) %>% arrange(Sample)
```

See the percentage of NKT gene segments in each sample:

```{r}
kable(nkt_table, format = "html") %>%
  kable_styling(full_width = F, font_size = 11, 
                bootstrap_options = c("striped", "hover", "condensed", "responsive"))

# source data
write.csv(nkt_table, "source_data/FigED7b_nkt_cells.csv", row.names = F)
```


Now we will plot the data for CD4 cells and for CD8 cells. Plots for TRA and TRB sequences are separated. 

## All NKT

This plot visualizes the percentage of any of the TRB NKT chains (TRBV1, TRBV13-2 and TRBV29). We can see that this distinction is not specific as these gene segments are also used by conventional CD4 and CD8 T cells. 

```{r fig.width=7, fig.height=3.5}
## CD4
p01 <- nkt_table_trb %>% 
  filter(Cell_type == "CD4") %>% 
  ggplot(aes(y = freq*100, x = factor(sample_type, levels = levels_cd4))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Mouse_strain)) + 
  ylab("Percentage of all chains") + 
  ggtitle("NKT segments in CD4 TRB") + 
  theme_classic() + 
  ggtheme() + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")
  
#ggsave("./final_fig/nkt/pct_all_nkt_trb_cd4.png", width = 8, height = 8, units = "cm")
#ggsave("./final_fig/nkt/pct_all_nkt_trb_cd4.eps", width = 8, height = 8, units = "cm")

## CD8
p02 <- nkt_table_trb %>% 
  filter(Cell_type == "CD8") %>% 
  ggplot(aes(y = freq*100, x = factor(sample_type, levels = levels_cd8))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Mouse_strain)) + 
  ylab("Percentage of all chains") + 
  ggtitle("NKT segments in CD8 TRB") + 
  theme_classic() + 
  ggtheme() + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")
  
#ggsave("./final_fig/nkt/pct_all_nkt_trb_cd8.png", width = 8, height = 8, units = "cm")
#ggsave("./final_fig/nkt/pct_all_nkt_trb_cd8.eps", width = 8, height = 8, units = "cm")

p01 + p02
```


### TRAV11 TRAJ18

```{r fig.width=7, fig.height=3.5}
## CD4
p03 <- nkt_table_trav11_traj18 %>% 
  filter(Cell_type == "CD4") %>% 
  ggplot(aes(y = freq*100, x = factor(sample_type, levels = levels_cd4))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Mouse_strain)) + 
  ylab("Percentage of all chains") + 
  ggtitle("TRAV11-TRAJ18 segments in CD4 cells") + 
  theme_classic() + 
  ggtheme() + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")
  
#ggsave("./final_fig/nkt/pct_nkt_trav11_traj18_tra_cd4.png", width = 8, height = 8, units = "cm")
#ggsave("./final_fig/nkt/pct_nkt_trav11_traj18_tra_cd4.eps", width = 8, height = 8, units = "cm")

## CD8
p04 <- nkt_table_trav11_traj18 %>% 
  filter(Cell_type == "CD8") %>% 
  ggplot(aes(y = freq*100, x = factor(sample_type, levels = levels_cd8))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Mouse_strain)) + 
  ylab("Percentage of all chains") + 
  ggtitle("TRAV11-TRAJ18 segments in CD8 cells") + 
  theme_classic() + 
  ggtheme() + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,1)) + xlab("")
  
#ggsave("./final_fig/nkt/pct_nkt_trav11_traj18_tra_cd8.png", width = 8, height = 8, units = "cm")
#ggsave("./final_fig/nkt/pct_nkt_trav11_traj18_tra_cd8.eps", width = 8, height = 8, units = "cm")

p03 + p04
```

### TRBV1

```{r fig.width=7, fig.height=3.5}

## CD4
p05 <- nkt_table_trbv1 %>% 
  filter(Cell_type == "CD4") %>% 
  ggplot(aes(y = freq*100, x = factor(sample_type, levels = levels_cd4))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Mouse_strain)) + 
  ylab("Percentage of all chains") + 
  ggtitle("CD4 TRBV1") + 
  theme_classic() + 
  ggtheme() + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")
  
#ggsave("./final_fig/nkt/pct_nkt_trbv1_cd4.png", width = 8, height = 8, units = "cm")
#ggsave("./final_fig/nkt/pct_nkt_trbv1_cd4.eps", width = 8, height = 8, units = "cm")

## CD8
p06 <- nkt_table_trbv1 %>% 
  filter(Cell_type == "CD8") %>% 
  ggplot(aes(y = freq*100, x = factor(sample_type, levels = levels_cd8))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Mouse_strain)) + 
  ylab("Percentage of all chains") + 
  ggtitle("CD8 TRBV1") + 
  theme_classic() + 
  ggtheme() + 
  
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")
  
#ggsave("./final_fig/nkt/pct_nkt_trbv1_cd8.png", width = 8, height = 8, units = "cm")
#ggsave("./final_fig/nkt/pct_nkt_trbv1_cd8.eps", width = 8, height = 8, units = "cm")

p05 + p06
```

### TRBV13-2

```{r fig.width=7, fig.height=3.5}

## CD4
p07 <- nkt_table_trbv13_2 %>% 
  filter(Cell_type == "CD4") %>% 
  ggplot(aes(y = freq*100, x = factor(sample_type, levels = levels_cd4))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Mouse_strain)) + 
  ylab("Percentage of all chains") + 
  ggtitle("TRBV13-2 segments in CD4 cells") + 
  theme_classic() + 
  ggtheme() + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")
  
#ggsave("./final_fig/nkt/pct_nkt_trbv13_2_cd4.png", width = 8, height = 8, units = "cm")
#ggsave("./final_fig/nkt/pct_nkt_trbv13_2_cd4.eps", width = 8, height = 8, units = "cm")

## CD8
p08 <- nkt_table_trbv13_2 %>% 
  filter(Cell_type == "CD8") %>% 
  ggplot(aes(y = freq*100, x = factor(sample_type, levels = levels_cd8))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Mouse_strain)) + 
  ylab("Percentage of all chains") + 
  ggtitle("TRBV13-2 segments in CD8 cells") + 
  theme_classic() + 
  ggtheme() + 
  
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")
  
#ggsave("./final_fig/nkt/pct_nkt_trbv13_2_cd8.png", width = 8, height = 8, units = "cm")
#ggsave("./final_fig/nkt/pct_nkt_trbv13_2_cd8.eps", width = 8, height = 8, units = "cm")

p07 + p08
```

### TRBV29

```{r fig.width=7, fig.height=3.5}

## CD4
p09 <- nkt_table_trbv29 %>% 
  filter(Cell_type == "CD4") %>% 
  ggplot(aes(y = freq*100, x = factor(sample_type, levels = levels_cd4))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Mouse_strain)) + 
  ylab("Percentage of all chains") + 
  ggtitle("CD4 TRBV29") + 
  theme_classic() + 
  ggtheme() + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")
  
#ggsave("./final_fig/nkt/pct_nkt_trbv29_cd4.png", width = 8, height = 8, units = "cm")
#ggsave("./final_fig/nkt/pct_nkt_trbv29_cd4.eps", width = 8, height = 8, units = "cm")

## CD8
p10 <- nkt_table_trbv29 %>% 
  filter(Cell_type == "CD8") %>% 
  ggplot(aes(y = freq*100, x = factor(sample_type, levels = levels_cd8))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Mouse_strain)) + 
  ylab("Percentage of all chains") + 
  ggtitle("CD8 TRBV29") + 
  theme_classic() + 
  ggtheme() + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")
  
#ggsave("./final_fig/nkt/pct_nkt_trbv29_cd8.png", width = 8, height = 8, units = "cm")
#ggsave("./final_fig/nkt/pct_nkt_trbv29_cd8.eps", width = 8, height = 8, units = "cm")

p09 + p10
```

# Principal component analysis

In this part we explore the similarity of TCR repertoires of different samples using principal component analysis (PCA).

We will calculate the distances from a filtered matrix of proportions, in which we filtered out the CDR3 sequences that are present in less than five CD4 or in less than five CD8 samples. 

## Preparing filtered normalized count matrices

First, we will prepare the filtered normalized count matrix. 

Below, we find and save the CDR3 TRA/TRB amino acid sequences, which are present in 5 or more samples out of 14 CD4 samples, or in 5 or more samples out of 14 CD8 samples. 

```{r}

# TRA
keep_tra_cd4 <- excel_count_table_tra %>% 
  mutate(nkt_trav11_traj18 = if_else(
    (grepl(allVHitsWithScore, pattern = "TRAV11") | 
    (grepl(allJHitsWithScore, pattern = "TRAJ18"))),"yes","no")) %>% 
  filter(nkt_trav11_traj18 == "no") %>% 
  select(aaSeqCDR3, starts_with("CD4")) %>% 
  mutate_at(vars(starts_with("CD4")), .funs = binary) 
 
keep_tra_cd4$sum <- rowSums((keep_tra_cd4 %>% select(-aaSeqCDR3)))

keep_tra_cd4$keep <- ifelse(keep_tra_cd4$sum>4,1,0)
keep_tra_cd4$orig <- ifelse(keep_tra_cd4$sum>0,1,0)

keep_tra_cd4_sequences <- pull(keep_tra_cd4 %>% filter(keep == 1), aaSeqCDR3)

# TRB
keep_trb_cd4 <- excel_count_table_trb %>% select(aaSeqCDR3, starts_with("CD4")) %>%  mutate_at(vars(starts_with("CD4")), .funs = binary) 
 
keep_trb_cd4$sum <- rowSums((keep_trb_cd4 %>% select(-aaSeqCDR3)))

keep_trb_cd4$keep <- ifelse(keep_trb_cd4$sum>4,1,0)
keep_trb_cd4$orig <- ifelse(keep_trb_cd4$sum>0,1,0)

keep_trb_cd4_sequences <- pull(keep_trb_cd4 %>% filter(keep == 1), aaSeqCDR3)

# TRA
keep_tra_cd8 <- excel_count_table_tra %>% 
  mutate(nkt_trav11_traj18 = if_else(
    (grepl(allVHitsWithScore, pattern = "TRAV11") | 
    (grepl(allJHitsWithScore, pattern = "TRAJ18"))),"yes","no")) %>% 
  filter(nkt_trav11_traj18 == "no") %>% 
  select(aaSeqCDR3, starts_with("CD8")) %>% 
  mutate_at(vars(starts_with("CD8")), .funs = binary) 

keep_tra_cd8$sum <- rowSums((keep_tra_cd8 %>% select(-aaSeqCDR3)))

keep_tra_cd8$keep <- ifelse(keep_tra_cd8$sum>4,1,0)
keep_tra_cd8$orig <- ifelse(keep_tra_cd8$sum>0,1,0)

keep_tra_cd8_sequences <- pull(keep_tra_cd8 %>% filter(keep == 1), aaSeqCDR3)

# TRB
keep_trb_cd8 <- excel_count_table_trb %>% select(aaSeqCDR3, starts_with("CD8")) %>%  mutate_at(vars(starts_with("CD8")), .funs = binary) 
 
keep_trb_cd8$sum <- rowSums((keep_trb_cd8 %>% select(-aaSeqCDR3)))

keep_trb_cd8$keep <- ifelse(keep_trb_cd8$sum>4,1,0)
keep_trb_cd8$orig <- ifelse(keep_trb_cd8$sum>0,1,0)

keep_trb_cd8_sequences <- pull(keep_trb_cd8 %>% filter(keep == 1), aaSeqCDR3)


cdr3_count_table <- data.frame(Sample = c("CD4 TRA","CD4 TRB","CD8 TRA","CD8 TRB"),
                               `Count before filtering` = c(
                                 length(pull(keep_tra_cd4 %>% filter(orig == 1), aaSeqCDR3)),
                                 length(pull(keep_trb_cd4 %>% filter(orig == 1), aaSeqCDR3)),
                                 length(pull(keep_tra_cd8 %>% filter(orig == 1), aaSeqCDR3)),
                                 length(pull(keep_trb_cd8 %>% filter(orig == 1), aaSeqCDR3))),
                               `Count after filtering` = c(
                                 length(pull(keep_tra_cd4 %>% filter(keep == 1), aaSeqCDR3)),
                                 length(pull(keep_trb_cd4 %>% filter(keep == 1), aaSeqCDR3)),
                                 length(pull(keep_tra_cd8 %>% filter(keep == 1), aaSeqCDR3)),
                                 length(pull(keep_trb_cd8 %>% filter(keep == 1), aaSeqCDR3))))

```

Here you can check the counts of CDR3 sequences before and after filtering. 

```{r}
kable(cdr3_count_table) %>%
  kable_styling(full_width = F, font_size = 11, 
                bootstrap_options = c("striped", "hover", "condensed", "responsive"))
```

Now we will create filtered tables with the abundant sequences.

```{r}

# TRB
excel_count_table_trb3_filter <- excel_count_table_trb2  %>% filter(aaSeqCDR3 %in% keep_trb_cd4_sequences | aaSeqCDR3 %in% keep_trb_cd8_sequences) 

count_table_trb4_filter <- as.matrix(excel_count_table_trb3_filter[,5:32])
rownames(count_table_trb4_filter) <- excel_count_table_trb3_filter$aaSeqCDR3

trb4_norm_filter <- scale(count_table_trb4_filter, center=FALSE, scale=colSums(count_table_trb4_filter))

prop.table.trb_filter <- cbind(trb4_norm_filter, excel_count_table_trb3_filter %>% select(-starts_with("CD")) )

# TRA
excel_count_table_tra3_filter <- excel_count_table_tra3  %>% filter(aaSeqCDR3 %in% keep_tra_cd4_sequences | aaSeqCDR3 %in% keep_tra_cd8_sequences) %>% filter(nkt_trav11_traj18 == "no")

count_table_tra4_filter <- as.matrix(excel_count_table_tra3_filter[,5:32])
rownames(count_table_tra4_filter) <- excel_count_table_tra3_filter$aaSeqCDR3

tra4_norm_filter <- scale(count_table_tra4_filter, center=FALSE, scale=colSums(count_table_tra4_filter))

prop.table.tra_filter <- cbind(tra4_norm_filter, excel_count_table_tra3_filter %>% select(-starts_with("CD")) )

colnames(prop.table.tra_filter)
```

Last, we will select the columns of the filtered table corresponding to CD4 and CD8 cells. We will remove the rows in which there are only zeros for the subset of CD4 or CD8 cells, respectively. 

```{r}
prop.table.tra2_filter.cd4 <- prop.table.tra_filter[,1:14]
prop.table.tra2_filter.cd4$sum <- rowSums(prop.table.tra2_filter.cd4)
prop.table.tra2_filter.cd4 <- prop.table.tra2_filter.cd4 %>% filter(sum >0) %>% select(-sum)

prop.table.trb2_filter.cd4 <- prop.table.trb_filter[,1:14]
prop.table.trb2_filter.cd4$sum <- rowSums(prop.table.trb2_filter.cd4)
prop.table.trb2_filter.cd4 <- prop.table.trb2_filter.cd4 %>% filter(sum >0) %>% select(-sum)

prop.table.tra2_filter.cd8 <- prop.table.tra_filter[,15:28]
prop.table.tra2_filter.cd8$sum <- rowSums(prop.table.tra2_filter.cd8)
prop.table.tra2_filter.cd8 <- prop.table.tra2_filter.cd8 %>% filter(sum >0) %>% select(-sum)

prop.table.trb2_filter.cd8 <- prop.table.trb_filter[,15:28]
prop.table.trb2_filter.cd8$sum <- rowSums(prop.table.trb2_filter.cd8)
prop.table.trb2_filter.cd8 <- prop.table.trb2_filter.cd8 %>% filter(sum >0) %>% select(-sum)
```

## PCA plots

Using the tables that we generated previously, we will compute the PCA with the package `factoextra`. First, we will merge TRA and TRB CDR3 sequences together and the we will also perform separated analyses. 

### TRA + TRB CD4


```{r fig.width=4, fig.height=3}
prop.table.tra2_filter.merge <- rbind(prop.table.tra2_filter.cd4, prop.table.trb2_filter.cd4)
  
res.pca.merge.cd4 <- prcomp(t(prop.table.tra2_filter.merge), scale = TRUE, center = T)

mdres.pca.merge.cd4  <-  colnames(prop.table.tra2_filter.merge[,1:14])  %>% 
  as.data.frame() %>% 
  mutate(sample = stringr::str_replace_all(., pattern = "Lymph nodes", replacement = "LN")) %>% 
  separate(sample, into = c("Cell_type","Organ","Strain","Exp"))

fviz_pca_ind(res.pca.merge.cd4,
             col.ind = as.factor(mdres.pca.merge.cd4$Strain), # color by groups
             legend.title = "Groups",
             geom.ind = "point",
             invisible="quali", pointsize = 0) +
   scale_color_manual(values=c("dodgerblue","indianred2","gray40")) +
   geom_point(aes(shape = as.factor(mdres.pca.merge.cd4$Organ),
                  color = as.factor(mdres.pca.merge.cd4$Strain))) + 
  scale_shape_manual(values = c(8,8,8,15,15)) +
  ggtheme() +
  theme(axis.text.x = element_text(angle = 0))


# source data
# write.csv(res.pca.merge.cd4$x, "source_data/pca_raw_cd4.csv")

#ggsave("final_fig/pca/cd4.png", width = 2.6, height = 1.7)
#ggsave("final_fig/pca/cd4.svg", width = 2.6, height = 1.7)
```

As the points in the central location are overlapping, we will also create a zoom plot. This will be added to the previous figure manually in a graphic editor.

```{r fig.width=4, fig.height=3}
fviz_pca_ind(res.pca.merge.cd4,
             col.ind = as.factor(mdres.pca.merge.cd4$Strain), # color by groups
             legend.title = "Groups",
             geom.ind = "point",
             invisible="quali", pointsize = 0) +
   scale_color_manual(values=c("dodgerblue","indianred2","gray40")) +
   geom_point(aes(shape = as.factor(mdres.pca.merge.cd4$Organ),
                  color = as.factor(mdres.pca.merge.cd4$Strain))) + 
  scale_shape_manual(values = c(8,8,8,15,15)) +
  xlim(c(-30,0)) + ylim(-2,15) +
  ggtheme() +
  theme(axis.text.x = element_text(angle = 0))


#ggsave("final_fig/pca/cd4_zoom.png", width = 2.2, height = 1.4)
#ggsave("final_fig/pca/cd4_zoom.svg", width = 2.2, height = 1.4)
```


### TRA + TRB CD8



```{r fig.width=4, fig.height=3}
prop.table.tra2_filter.merge <- rbind(prop.table.tra2_filter.cd8, prop.table.trb2_filter.cd8)
  
res.pca.merge.cd8 <- prcomp(t(prop.table.tra2_filter.merge), scale = TRUE, center = T)

mdres.pca.merge.cd8  <-  colnames(prop.table.tra2_filter.merge[,1:14])  %>% 
  as.data.frame() %>% 
  mutate(sample = stringr::str_replace_all(., pattern = "Lymph nodes", replacement = "LN")) %>% 
  separate(sample, into = c("Cell_type","Organ","Strain","Exp"))
  

fviz_pca_ind(res.pca.merge.cd8,
             col.ind = as.factor(mdres.pca.merge.cd8$Strain), # color by groups
             legend.title = "Groups",
             geom.ind = "point",
             invisible="quali", pointsize = 0) +
   scale_color_manual(values=c("dodgerblue","indianred2","gray40")) +
   geom_point(aes(shape = as.factor(mdres.pca.merge.cd8$Organ),
                  color = as.factor(mdres.pca.merge.cd8$Strain))) + 
  scale_shape_manual(values = c(8,8,8,15,15)) +
  ggtheme()

# source data
write.csv(res.pca.merge.cd8$x, "source_data/pca_raw_cd8.csv")

#ggsave("final_fig/pca/cd8.png", width = 2.6, height = 1.7)
#ggsave("final_fig/pca/cd8.svg", width = 2.6, height = 1.7)
```

As the points in the central location are overlapping, we will also create a zoom plot. This will be added to the previous figure manually in a graphic editor.


```{r fig.width=4, fig.height=3}
fviz_pca_ind(res.pca.merge.cd8,
             col.ind = as.factor(mdres.pca.merge.cd8$Strain), # color by groups
             legend.title = "Groups",
             geom.ind = "point",
             invisible="quali", pointsize = 0) +
   scale_color_manual(values=c("dodgerblue","indianred2","gray40")) +
   geom_point(aes(shape = as.factor(mdres.pca.merge.cd8$Organ),
                  color = as.factor(mdres.pca.merge.cd8$Strain))) + 
  scale_shape_manual(values = c(8,8,8,15,15)) +
  xlim(c(-32,-18)) + ylim(-4,5) +
  ggtheme()

#ggsave("final_fig/pca/cd8_zoom.png", width = 2.2, height = 1.4)
#ggsave("final_fig/pca/cd8_zoom.svg", width = 2.2, height = 1.4)
```

Next, we will create eight separate plots which show the PCA for TRA/TRB CDR3 sequences from each cell type and each organ. 

### TRA CD4 Thymus

```{r fig.width=4, fig.height=3}
## Thymus
prop.table.tra2_filter.cd4.thy <- prop.table.tra_filter[,8:14]
prop.table.tra2_filter.cd4.thy$sum <- rowSums(prop.table.tra2_filter.cd4.thy)

prop.table.tra2_filter.cd4.thy <- prop.table.tra2_filter.cd4.thy %>% filter(sum >0) %>% select(-sum)

res.pca.tra.cd4.thy <- prcomp(t(prop.table.tra2_filter.cd4.thy), scale = TRUE, center = T)

mdres.pca.tra.cd4.thy  <-  colnames(prop.table.tra_filter[,8:14])  %>% 
  as.data.frame() %>% 
  mutate(sample = stringr::str_replace_all(., pattern = "Lymph nodes", replacement = "LN")) %>% 
  separate(sample, into = c("Cell_type","Organ","Strain","Exp"))
  
fviz_pca_ind(res.pca.tra.cd4.thy,
             col.ind = as.factor(mdres.pca.tra.cd4.thy$Strain), # color by groups
             legend.title = "Groups",
             geom.ind = "point",
             invisible="quali", pointsize = 0) +
   scale_color_manual(values=c("dodgerblue","indianred2","gray40")) +
   geom_point(aes(shape = as.factor(mdres.pca.tra.cd4.thy$Organ),
                  color = as.factor(mdres.pca.tra.cd4.thy$Strain))) +
  geom_hline(yintercept = 0, color = "grey20") +
  geom_vline(xintercept = 0, color = "grey20") +
  ggtheme()

#ggsave("final_fig/pca/cd4_tra_thy.png", width = 2.6, height = 1.7)
#ggsave("final_fig/pca/cd4_tra_thy.svg", width = 2.6, height = 1.7)

```

### TRA CD4 Lymph nodes

```{r fig.width=4, fig.height=3}
## LN
prop.table.tra2_filter.cd4.ln <- prop.table.tra_filter[,1:7]
prop.table.tra2_filter.cd4.ln$sum <- rowSums(prop.table.tra2_filter.cd4.ln)

prop.table.tra2_filter.cd4.ln <- prop.table.tra2_filter.cd4.ln %>% filter(sum >0) %>% select(-sum)

res.pca.tra.cd4.ln <- prcomp(t(prop.table.tra2_filter.cd4.ln), scale = TRUE, center = T)

mdres.pca.tra.cd4.ln  <-  colnames(prop.table.tra_filter[,1:7])  %>% 
  as.data.frame() %>% 
  mutate(sample = stringr::str_replace_all(., pattern = "Lymph nodes", replacement = "LN")) %>% 
  separate(sample, into = c("Cell_type","Organ","Strain","Exp"))
  

fviz_pca_ind(res.pca.tra.cd4.ln,
             col.ind = as.factor(mdres.pca.tra.cd4.ln$Strain), # color by groups
             legend.title = "Groups",
             geom.ind = "point",
             invisible="quali", pointsize = 0) +
   scale_color_manual(values=c("dodgerblue","indianred2","gray40")) +
   geom_point(aes(shape = as.factor(mdres.pca.tra.cd4.ln$Organ),
                  color = as.factor(mdres.pca.tra.cd4.ln$Strain))) +
  geom_hline(yintercept = 0, color = "grey20") +
  geom_vline(xintercept = 0, color = "grey20") 

#ggsave("final_fig/pca/cd4_tra_ln.png", width = 2.6, height = 1.7)
#ggsave("final_fig/pca/cd4_tra_ln.svg", width = 2.6, height = 1.7)
```

### TRA CD8 Thymus

```{r fig.width=4, fig.height=3}
## Thymus
prop.table.tra2_filter %>% colnames
prop.table.tra2_filter.cd8.thy <- prop.table.tra_filter[,22:28]
prop.table.tra2_filter.cd8.thy$sum <- rowSums(prop.table.tra2_filter.cd8.thy)

prop.table.tra2_filter.cd8.thy <- prop.table.tra2_filter.cd8.thy %>% filter(sum >0) %>% select(-sum)

res.pca.tra.cd8.thy <- prcomp(t(prop.table.tra2_filter.cd8.thy), scale = TRUE, center = T)

mdres.pca.tra.cd8.thy  <-  colnames(prop.table.tra_filter[,22:28])  %>% 
  as.data.frame() %>% 
  mutate(sample = stringr::str_replace_all(., pattern = "Lymph nodes", replacement = "LN")) %>% 
  separate(sample, into = c("Cell_type","Organ","Strain","Exp"))
 
fviz_pca_ind(res.pca.tra.cd8.thy,
             col.ind = as.factor(mdres.pca.tra.cd8.thy$Strain), # color by groups
             legend.title = "Groups",
             geom.ind = "point",
             invisible="quali", pointsize = 0) +
   scale_color_manual(values=c("dodgerblue","indianred2","gray40")) +
   geom_point(aes(shape = as.factor(mdres.pca.tra.cd8.thy$Organ),
                  color = as.factor(mdres.pca.tra.cd8.thy$Strain))) +
  geom_hline(yintercept = 0, color = "grey20") +
  geom_vline(xintercept = 0, color = "grey20") 

#ggsave("final_fig/pca/cd8_tra_thy.png", width = 2.6, height = 1.7)
#ggsave("final_fig/pca/cd8_tra_thy.svg", width = 2.6, height = 1.7)
```

### TRA CD8 Lymph nodes

```{r fig.width=4, fig.height=3}
## LN
prop.table.tra2_filter %>% colnames
prop.table.tra2_filter.cd8.ln <- prop.table.tra_filter[,15:21]
prop.table.tra2_filter.cd8.ln$sum <- rowSums(prop.table.tra2_filter.cd8.ln)

prop.table.tra2_filter.cd8.ln <- prop.table.tra2_filter.cd8.ln %>% filter(sum >0) %>% select(-sum)

res.pca.tra.cd8.ln <- prcomp(t(prop.table.tra2_filter.cd8.ln), scale = TRUE, center = T)

mdres.pca.tra.cd8.ln  <-  colnames(prop.table.tra_filter[,15:21])  %>% 
  as.data.frame() %>% 
  mutate(sample = stringr::str_replace_all(., pattern = "Lymph nodes", replacement = "LN")) %>% 
  separate(sample, into = c("Cell_type","Organ","Strain","Exp"))
  
fviz_pca_ind(res.pca.tra.cd8.ln,
             col.ind = as.factor(mdres.pca.tra.cd8.ln$Strain), # color by groups
             legend.title = "Groups",
             geom.ind = "point",
             invisible="quali", pointsize = 0) +
   scale_color_manual(values=c("dodgerblue","indianred2","gray40")) +
   geom_point(aes(shape = as.factor(mdres.pca.tra.cd8.ln$Organ),
                  color = as.factor(mdres.pca.tra.cd8.ln$Strain))) +
  geom_hline(yintercept = 0, color = "grey20") +
  geom_vline(xintercept = 0, color = "grey20") 

#ggsave("final_fig/pca/cd8_tra_ln.png", width = 2.6, height = 1.7)
#ggsave("final_fig/pca/cd8_tra_ln.svg", width = 2.6, height = 1.7)
```

### TRB CD4 Thymus

```{r fig.width=4, fig.height=3}
## Thymus
prop.table.trb_filter %>% colnames
prop.table.trb2_filter.cd4.thy <- prop.table.trb_filter[,12:18]
prop.table.trb2_filter.cd4.thy$sum <- rowSums(prop.table.trb2_filter.cd4.thy)

prop.table.trb2_filter.cd4.thy <- prop.table.trb2_filter.cd4.thy %>% filter(sum >0) %>% select(-sum)

res.pca.trb.cd4.thy <- prcomp(t(prop.table.trb2_filter.cd4.thy), scale = TRUE, center = T)

mdres.pca.trb.cd4.thy  <-  colnames(prop.table.trb_filter[,12:18])  %>% 
  as.data.frame() %>% 
  mutate(sample = stringr::str_replace_all(., pattern = "Lymph nodes", replacement = "LN")) %>% 
  separate(sample, into = c("Cell_type","Organ","Strain","Exp"))
 
fviz_pca_ind(res.pca.trb.cd4.thy,
             col.ind = as.factor(mdres.pca.trb.cd4.thy$Strain), # color by groups
             legend.title = "Groups",
             geom.ind = "point",
             invisible="quali", pointsize = 0) +
   scale_color_manual(values=c("dodgerblue","indianred2","gray40")) +
   geom_point(aes(shape = as.factor(mdres.pca.trb.cd4.thy$Organ),
                  color = as.factor(mdres.pca.trb.cd4.thy$Strain))) +
  geom_hline(yintercept = 0, color = "grey20") +
  geom_vline(xintercept = 0, color = "grey20") 
#ggsave("final_fig/pca/cd4_trb_thy.png", width = 2.6, height = 1.7)
#ggsave("final_fig/pca/cd4_trb_thy.svg", width = 2.6, height = 1.7)

```

### TRB CD4 Lymph nodes

```{r fig.width=4, fig.height=3}
## All
colnames(prop.table.trb2_filter)

prop.table.trb2_filter.cd4 <- prop.table.trb_filter[,5:18]
prop.table.trb2_filter.cd4$sum <- rowSums(prop.table.trb2_filter.cd4)

prop.table.trb2_filter.cd4 <- prop.table.trb2_filter.cd4 %>% filter(sum >0) %>% select(-sum)

res.pca.trb.cd4 <- prcomp(t(prop.table.trb2_filter.cd4), scale = TRUE, center = T)

mdres.pca.trb.cd4  <-  colnames(prop.table.trb_filter[,5:18])  %>% 
  as.data.frame() %>% 
  mutate(sample = stringr::str_replace_all(., pattern = "Lymph nodes", replacement = "LN")) %>% 
  separate(sample, into = c("Cell_type","Organ","Strain","Exp"))
 

## LN
prop.table.trb_filter %>% colnames
prop.table.trb2_filter.cd4.ln <- prop.table.trb_filter[,5:11]
prop.table.trb2_filter.cd4.ln$sum <- rowSums(prop.table.trb2_filter.cd4.ln)

prop.table.trb2_filter.cd4.ln <- prop.table.trb2_filter.cd4.ln %>% filter(sum >0) %>% select(-sum)

res.pca.trb.cd4.ln <- prcomp(t(prop.table.trb2_filter.cd4.ln), scale = TRUE, center = T)

mdres.pca.trb.cd4.ln  <-  colnames(prop.table.trb_filter[,5:11])  %>% 
  as.data.frame() %>% 
  mutate(sample = stringr::str_replace_all(., pattern = "Lymph nodes", replacement = "LN")) %>% 
  separate(sample, into = c("Cell_type","Organ","Strain","Exp"))
  

fviz_pca_ind(res.pca.trb.cd4.ln,
             col.ind = as.factor(mdres.pca.trb.cd4.ln$Strain), # color by groups
             legend.title = "Groups",
             geom.ind = "point",
             invisible="quali", pointsize = 0) +
   scale_color_manual(values=c("dodgerblue","indianred2","gray40")) +
   geom_point(aes(shape = as.factor(mdres.pca.trb.cd4.ln$Organ),
                  color = as.factor(mdres.pca.trb.cd4.ln$Strain))) +
  geom_hline(yintercept = 0, color = "grey20") +
  geom_vline(xintercept = 0, color = "grey20") 
#ggsave("final_fig/pca/cd4_trb_ln.png", width = 2.6, height = 1.7)
#ggsave("final_fig/pca/cd4_trb_ln.svg", width = 2.6, height = 1.7)
```

### TRB CD8 Thymus

```{r fig.width=4, fig.height=3}

## Thymus
prop.table.trb_filter %>% colnames
prop.table.trb2_filter.cd8.thy <- prop.table.trb_filter[,26:32]
prop.table.trb2_filter.cd8.thy$sum <- rowSums(prop.table.trb2_filter.cd8.thy)

prop.table.trb2_filter.cd8.thy <- prop.table.trb2_filter.cd8.thy %>% filter(sum >0) %>% select(-sum)

res.pca.trb.cd8.thy <- prcomp(t(prop.table.trb2_filter.cd8.thy), scale = TRUE, center = T)

mdres.pca.trb.cd8.thy  <-  colnames(prop.table.trb_filter[,26:32])  %>% 
  as.data.frame() %>% 
  mutate(sample = stringr::str_replace_all(., pattern = "Lymph nodes", replacement = "LN")) %>% 
  separate(sample, into = c("Cell_type","Organ","Strain","Exp"))
 
fviz_pca_ind(res.pca.trb.cd8.thy,
             col.ind = as.factor(mdres.pca.trb.cd8.thy$Strain), # color by groups
             legend.title = "Groups",
             geom.ind = "point",
             invisible="quali", pointsize = 0) +
   scale_color_manual(values=c("dodgerblue","indianred2","gray40")) +
   geom_point(aes(shape = as.factor(mdres.pca.trb.cd8.thy$Organ),
                  color = as.factor(mdres.pca.trb.cd8.thy$Strain))) +
  geom_hline(yintercept = 0, color = "grey20") +
  geom_vline(xintercept = 0, color = "grey20") 
#ggsave("final_fig/pca/cd8_trb_thy.png", width = 2.6, height = 1.7)
#ggsave("final_fig/pca/cd8_trb_thy.svg", width = 2.6, height = 1.7)

```

### TRB CD8 Lymph nodes

```{r fig.width=4, fig.height=3}
## LN
prop.table.trb_filter %>% colnames
prop.table.trb2_filter.cd8.ln <- prop.table.trb_filter[,19:25]
prop.table.trb2_filter.cd8.ln$sum <- rowSums(prop.table.trb2_filter.cd8.ln)

prop.table.trb2_filter.cd8.ln <- prop.table.trb2_filter.cd8.ln %>% filter(sum >0) %>% select(-sum)

res.pca.trb.cd8.ln <- prcomp(t(prop.table.trb2_filter.cd8.ln), scale = TRUE, center = T)

mdres.pca.trb.cd8.ln  <-  colnames(prop.table.trb_filter[,19:25])  %>% 
  as.data.frame() %>% 
  mutate(sample = stringr::str_replace_all(., pattern = "Lymph nodes", replacement = "LN")) %>% 
  separate(sample, into = c("Cell_type","Organ","Strain","Exp"))
  
fviz_pca_ind(res.pca.trb.cd8.ln,
             col.ind = as.factor(mdres.pca.trb.cd8.ln$Strain), # color by groups
             legend.title = "Groups",
             geom.ind = "point",
             invisible="quali", pointsize = 0) +
   scale_color_manual(values=c("dodgerblue","indianred2","gray40")) +
   geom_point(aes(shape = as.factor(mdres.pca.trb.cd8.ln$Organ),
                  color = as.factor(mdres.pca.trb.cd8.ln$Strain))) +
  geom_hline(yintercept = 0, color = "grey20") +
  geom_vline(xintercept = 0, color = "grey20") 
#ggsave("final_fig/pca/cd8_trb_ln.png", width = 2.6, height = 1.7)
#ggsave("final_fig/pca/cd8_trb_ln.svg", width = 2.6, height = 1.7)
```

# Diversity index
 To compare the diversity of the TCR repertoires of our samples, we will calculate the Chao1 index using the `Immunarch` package. Chao1 is a nonparameteric asymptotic estimator of species richness. The higher the Chao1 value, the more diverse repertoire in the sample. 

We will calculate the diversity index of TCRa chains without iNKT chains, so we will first exclude them.

```{r}
# Read metadata
md <- read.csv("metadata_Lck.csv")

# Create filtered dataset
immdata_tra_wonkt <- repFilter(immdata_tra, "by.clonotype",
  list(V.name = exclude("TRAV11"), J.name = exclude("TRAJ18")),
  .match = "substring")
```


```{r}
## ALL TRA
repDiversity(immdata_tra_wonkt$data) %>% vis(.by = c( "Mouse_strain"), .meta = immdata_tra$meta, 
  .signif.label.size = 0) + ggtitle("TRA without NKT") + scale_fill_manual(values = c("dodgerblue","indianred2","gray70")) + ggtitle("all TRA", subtitle = "")

# source data
write.csv(repDiversity(immdata_tra_wonkt$data), "rep_diversity_tra.csv")

#ggsave("./final_fig/diversity/chao_tra_wonkt_sign.png", width = 8, height = 13, units = "cm")
#ggsave("./final_fig/diversity/chao_tra_wonkt_sign.svg", width = 8, height = 13, units = "cm")
```

The TCRb repertoire doesn't need any filtering, so we can use it as it is.

```{r}
## ALL TRB
repDiversity(immdata_trb$data) %>% vis(.by = c( "Mouse_strain"), .meta = immdata_trb$meta, 
  .signif.label.size = 0) + ggtitle("TRB without NKT") + scale_fill_manual(values = c("dodgerblue","indianred2","gray70")) + ggtitle("all TRB", subtitle = "")

# source data
write.csv(repDiversity(immdata_trb$data), "rep_diversity_trb.csv")

#ggsave("./final_fig/diversity/chao_trb_sign.png", width = 8, height = 13, units = "cm")
#ggsave("./final_fig/diversity/chao_trb_sign.svg", width = 8, height = 13, units = "cm")
```

Next, we divide samples by tissue and by cell type (CD4 versus CD8).

```{r}
## Chao index
repDiversity(immdata_tra_wonkt$data) %>% vis(.by = c("Cell_type", "Organ","Mouse_strain"), .meta = immdata_tra$meta, .test = F,
  .signif.label.size = 0) + ggtitle("TRA")
#ggsave("./plots/chao_tra.png", width = 15, height = 11, units = "cm")
#ggsave("./plots/chao_tra.svg", width = 15, height = 11, units = "cm")

repDiversity(immdata_trb$data) %>% vis(.by = c("Cell_type", "Organ","Mouse_strain"), .meta = immdata_trb$meta, .test = F,
  .signif.label.size = 0) + ggtitle("TRB", subtitle = "")
#ggsave("./plots/chao_trb.png", width = 15, height = 11, units = "cm")
#ggsave("./plots/chao_trb.svg", width = 15, height = 11, units = "cm")

```

```{r}
chao_tra <- as.data.frame(repDiversity(immdata_tra_wonkt$data))
chao_trb <- as.data.frame(repDiversity(immdata_trb$data))
```


```{r}

chao_tra2 <- chao_tra %>% 
  rownames_to_column("num_id") %>% 
mutate(num_id = str_replace_all(string = num_id, c("new_lib_24" = "lib24_S24_L001",  "new_lib_25" = "lib25_S25_L001"))) %>% 
  separate(num_id, into = c("num_id",NA,NA), sep = "_") %>% 
  mutate(num_id = str_replace(num_id, "lib","")) %>% 
  mutate_at(vars("num_id"), as.numeric) 
  

levels_cd4 <- c("CD4 Thymus WT", "CD4 Thymus CA", "CD4 Thymus CAKR", "CD4 Lymph nodes WT", "CD4 Lymph nodes CA", "CD4 Lymph nodes CAKR")

# Attach metadata sent for sequencing

chao_tra2 <- left_join(chao_tra2, md) %>% select(value = Estimator, Cell_type, Organ, Mouse_strain) %>%
   mutate(sample_type = paste(Cell_type, Organ, Mouse_strain)) 

chao_tra2 %>% filter(Cell_type == "CD4") %>% 
  ggplot(aes(y = value, x = factor(sample_type, levels = levels_cd4))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Mouse_strain)) + 
  ylab("Chao index estimate") + 
  ggtitle("CD4 TRA") + 
  theme(axis.text.x = element_text(angle = 90)) + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) +
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")
  
#ggsave("./final_fig/diversity/chao_cd4_tra.png", width = 8, height = 10, units = "cm")
#ggsave("./final_fig/diversity/chao_cd4_tra.eps", width = 8, height = 10, units = "cm")

levels_cd8 <- c("CD8 Thymus WT", "CD8 Thymus CA", "CD8 Thymus CAKR", "CD8 Lymph nodes WT", "CD8 Lymph nodes CA", "CD8 Lymph nodes CAKR")

chao_tra2 %>% filter(Cell_type == "CD8") %>% 
  ggplot(aes(y = value, x = factor(sample_type, levels = levels_cd8))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Mouse_strain)) + 
  ylab("Chao index estimate") + 
  ggtitle("CD8 TRA") + 
  theme(axis.text.x = element_text(angle = 90)) + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) +
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")
  
#ggsave("./final_fig/diversity/chao_cd8_tra.png", width = 8, height = 10, units = "cm")
#ggsave("./final_fig/diversity/chao_cd8_tra.eps", width = 8, height = 10, units = "cm")

```


```{r}

chao_trb2 <- chao_trb %>% 
  rownames_to_column("num_id") %>% mutate(num_id = str_replace_all(string = num_id, c("new_lib_24" = "lib24_S24_L001",  "new_lib_25" = "lib25_S25_L001"))) %>% 
  separate(num_id, into = c("num_id",NA,NA), sep = "_") %>% 
  mutate(num_id = str_replace(num_id, "lib",""))  %>% 
  mutate_at(vars("num_id"), as.numeric) 
  

levels_cd4 <- c("CD4 Thymus WT", "CD4 Thymus CA", "CD4 Thymus CAKR", "CD4 Lymph nodes WT", "CD4 Lymph nodes CA", "CD4 Lymph nodes CAKR")

chao_trb2 <- left_join(chao_trb2, md) %>% select(value = Estimator, Cell_type, Organ, Mouse_strain) %>%
   mutate(sample_type = paste(Cell_type, Organ, Mouse_strain)) 

chao_trb2 %>% filter(Cell_type == "CD4") %>% 
  ggplot(aes(y = value, x = factor(sample_type, levels = levels_cd4))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Mouse_strain)) + 
  ylab("Chao index estimate") + 
  ggtitle("CD4 TRB") + 
  theme(axis.text.x = element_text(angle = 90)) + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) +
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,125500)) + xlab("")
  
#ggsave("./final_fig/diversity/chao_cd4_trb.png", width = 8, height = 10, units = "cm")
#ggsave("./final_fig/diversity/chao_cd4_trb.eps", width = 8, height = 10, units = "cm")

levels_cd8 <- c("CD8 Thymus WT", "CD8 Thymus CA", "CD8 Thymus CAKR", "CD8 Lymph nodes WT", "CD8 Lymph nodes CA", "CD8 Lymph nodes CAKR")


chao_trb2 %>% filter(Cell_type == "CD8") %>% 
  ggplot(aes(y = value, x = factor(sample_type, levels = levels_cd8))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Mouse_strain)) + 
  ylab("Chao index estimate") + 
  ggtitle("CD8 TRB") + 
  theme(axis.text.x = element_text(angle = 90)) + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) +
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,125500)) + xlab("")
  
#ggsave("./final_fig/diversity/chao_cd8_trb.png", width = 8, height = 10, units = "cm")
#ggsave("./final_fig/diversity/chao_cd8_trb.eps", width = 8, height = 10, units = "cm")

```


```{r}
chao_trb2 %>% 
  ggplot(aes(y = value, x = factor(Mouse_strain, levels = c("WT","CA","CAKR")))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=0) + 
  geom_jitter(position=position_jitter(0.1), size = 2, aes(color = Mouse_strain)) +
ylab("Chao index estimate") + 
  theme(axis.text.x = element_text(angle = 90)) + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) +
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")
  
#ggsave("./final_fig/diversity/chao_all_trb.png", width = 8, height = 6, units = "cm")
#ggsave("./final_fig/diversity/chao_all_trb.eps", width = 8, height = 6, units = "cm")


chao_tra2 %>% 
  ggplot(aes(y = value, x = factor(Mouse_strain, levels = c("WT","CA","CAKR")))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=0) + 
  geom_jitter(position=position_jitter(0.1), size = 2, aes(color = Mouse_strain)) +
ylab("Chao index estimate") + 
  theme(axis.text.x = element_text(angle = 90)) + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) +
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")
  
#ggsave("./final_fig/diversity/chao_all_tra.png", width = 8, height = 6, units = "cm")
#ggsave("./final_fig/diversity/chao_all_tra.eps", width = 8, height = 6, units = "cm")

```

# Top20 WT LN sequences

In the next analyses, we will figure out whether the sequences that are abundant in the periphery of WT animals are also present in LckKO animals. We will select the top 20 most abundant sequences from the lymph nodes of WT animals by calculating the average normalized frequency:

```{r}
## TRA
topclones <- prop.table.tra_filter %>% 
  mutate(topclones_ln_cd4 = (`CD4 Lymph nodes WT Exp02`+
                               `CD4 Lymph nodes WT Exp03`)/2,
         topclones_ln_cd8 = (`CD8 Lymph nodes WT Exp01`+
                               `CD8 Lymph nodes WT Exp02`+
                               `CD8 Lymph nodes WT Exp03`)/3)

cd4_topclones_20sequences_tra <-
  pull(topclones %>% slice_max(order_by = topclones_ln_cd4, n = 20),
       aaSeqCDR3)
cd8_topclones_20sequences_tra <-
  pull(topclones %>% slice_max(order_by = topclones_ln_cd8, n = 20),
       aaSeqCDR3)

## TRB
topclones <- prop.table.trb_filter %>% 
  mutate(topclones_ln_cd4 = (`CD4 Lymph nodes WT Exp02`+
                               `CD4 Lymph nodes WT Exp03`)/2,
         topclones_ln_cd8 = (`CD8 Lymph nodes WT Exp01`+
                               `CD8 Lymph nodes WT Exp02`+
                               `CD8 Lymph nodes WT Exp03`)/3)

cd4_topclones_20sequences_trb <- pull(topclones %>% 
                                        slice_max(order_by = topclones_ln_cd4, n = 20), aaSeqCDR3)
cd8_topclones_20sequences_trb <- pull(topclones %>% 
                                        slice_max(order_by = topclones_ln_cd8, n = 20), aaSeqCDR3)
```

Top 20 most abundant TRA sequences in CD4 WT LNs:

```{r}
cd4_topclones_20sequences_tra
```

Top 20 most abundant TRB sequences in CD4 WT LNs:

```{r}
cd4_topclones_20sequences_trb
```

Top 20 most abundant TRA sequences in CD8 WT LNs:

```{r}
cd8_topclones_20sequences_tra
```

Top 20 most abundant TRB sequences in CD8 WT LNs:

```{r}
cd8_topclones_20sequences_trb
```

## Summary plots - percentage of the repertoire

Now we will investigate what is the percentage of the repertoire of each sample that is occupied by the top20 WT LN sequences that we identified earlier. We will plot the results.

### CD4 TRA

```{r fig.width=3, fig.height=3}
cd4_topclones_rep_pct <- prop.table.tra2 %>% 
  filter(aaSeqCDR3 %in% cd4_topclones_20sequences_tra) %>% 
  select(aaSeqCDR3, starts_with("CD4")) %>% 
  select(1,14,15,9:13,7,8,2:6) 

cd4_topclones_rep_pct2 <- as.data.frame(colSums(cd4_topclones_rep_pct[,2:15]))
colnames(cd4_topclones_rep_pct2) <- "value"

cd4_topclones_rep_pct2 <- cd4_topclones_rep_pct2 %>% 
  rownames_to_column("sample") %>% 
  mutate(sample = stringr::str_replace_all(sample, pattern = "Lymph nodes", replacement = "LN")) %>% 
  separate(sample, into = c("Cell_type","Organ","Strain","Exp")) %>% 
  mutate(Cell_type_organ = factor(paste(Organ, Strain), 
                                  levels = c("Thymus WT", "Thymus CA", "Thymus CAKR",
                                             "LN WT", "LN CA", "LN CAKR")))

cd4_topclones_rep_pct2 %>% 
  filter(Organ == "LN") %>% 
  ggplot(aes(y = value*100, x = factor(Cell_type_organ))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Strain)) + 
  ylab("% of repertoire") + 
  ggtitle("CD4 TRA") + 
  theme(axis.text.x = element_text(angle = 90)) + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) +
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")

# source data
write.csv(cd4_topclones_rep_pct2 %>% 
  filter(Organ == "LN") %>%
  mutate(value = value*100), 
  file = "topclones_cd4_tra.csv")

#ggsave("./final_fig/20topseqs/20topseqs_pct_tra_cd4.png", width = 5, height = 5.7, units = "cm")
#ggsave("./final_fig/20topseqs/20topseqs_pct_tra_cd4.eps", width = 5, height = 5.7, units = "cm")

```

### CD4 TRB

```{r fig.width=3, fig.height=3}
cd4_topclones_rep_pct <- prop.table.trb2 %>% 
  filter(aaSeqCDR3 %in% cd4_topclones_20sequences_trb) %>% 
  select(aaSeqCDR3, starts_with("CD4")) %>% 
  select(1,14,15,9:13,7,8,2:6) 

cd4_topclones_rep_pct2 <- as.data.frame(colSums(cd4_topclones_rep_pct[,2:15]))
colnames(cd4_topclones_rep_pct2) <- "value"

cd4_topclones_rep_pct2 <- cd4_topclones_rep_pct2 %>% 
  rownames_to_column("sample") %>% 
  mutate(sample = stringr::str_replace_all(sample, pattern = "Lymph nodes", replacement = "LN")) %>% 
  separate(sample, into = c("Cell_type","Organ","Strain","Exp")) %>% 
  mutate(Cell_type_organ = factor(paste(Organ, Strain), levels = c("Thymus WT","Thymus CA", "Thymus CAKR",
                                                                   "LN WT","LN CA", "LN CAKR" )))

cd4_topclones_rep_pct2 %>% 
  filter(Organ == "LN") %>% 
  ggplot(aes(y = value*100, x = factor(Cell_type_organ))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Strain)) + 
  ylab("% of repertoire") + 
  ggtitle("CD4 TRB") + 
  theme(axis.text.x = element_text(angle = 90)) + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) +
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")

# source data
write.csv(cd4_topclones_rep_pct2 %>% 
  filter(Organ == "LN") %>%
  mutate(value = value*100), 
  file = "topclones_cd4_trb.csv")

#ggsave("./final_fig/20topseqs/20topseqs_pct_trb_cd4.png", width = 5.4, height = 5.7, units = "cm")
#ggsave("./final_fig/20topseqs/20topseqs_pct_trb_cd4.eps", width = 5.4, height = 5.7, units = "cm")

```

### CD8 TRA

```{r fig.width=3, fig.height=3}
cd8_topclones_rep_pct <- prop.table.tra2 %>% 
  filter(aaSeqCDR3 %in% cd8_topclones_20sequences_tra) %>% 
  select(aaSeqCDR3, starts_with("CD8")) %>% 
  select(1,13:15,9:12,6:8,2:5) 

cd8_topclones_rep_pct2 <- as.data.frame(colSums(cd8_topclones_rep_pct[,2:15]))
colnames(cd8_topclones_rep_pct2) <- "value"

cd8_topclones_rep_pct2 <- cd8_topclones_rep_pct2 %>% 
  rownames_to_column("sample") %>% 
  mutate(sample = stringr::str_replace_all(sample, pattern = "Lymph nodes", replacement = "LN")) %>% 
  separate(sample, into = c("Cell_type","Organ","Strain","Exp")) %>% 
  mutate(Cell_type_organ = factor(paste(Organ, Strain), levels = c("Thymus WT","Thymus CA", "Thymus CAKR",
                                                                   "LN WT","LN CA", "LN CAKR" )))

cd8_topclones_rep_pct2

cd8_topclones_rep_pct2 %>% 
  filter(Organ == "LN") %>% 
  ggplot(aes(y = value*100, x = factor(Cell_type_organ))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Strain)) + 
  ylab("% of repertoire") + 
  ggtitle("CD8 TRA") + 
  theme(axis.text.x = element_text(angle = 90)) + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) +
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")

# source data
write.csv(cd8_topclones_rep_pct2 %>% 
  filter(Organ == "LN") %>%
  mutate(value = value*100), 
  file = "topclones_cd8_tra.csv")

#ggsave("./final_fig/20topseqs/20topseqs_pct_tra_cd8.png", width = 5, height = 5.7, units = "cm")
#ggsave("./final_fig/20topseqs/20topseqs_pct_tra_cd8.eps", width = 5, height = 5.7, units = "cm")

```

### CD8 TRB
```{r fig.width=3, fig.height=3}
cd8_topclones_rep_pct <- prop.table.trb2 %>% 
  filter(aaSeqCDR3 %in% cd8_topclones_20sequences_trb) %>% 
  select(aaSeqCDR3, starts_with("CD8")) %>% 
  select(1,13:15,9:12,6:8,2:5) 

cd8_topclones_rep_pct2 <- as.data.frame(colSums(cd8_topclones_rep_pct[,2:15]))
colnames(cd8_topclones_rep_pct2) <- "value"

cd8_topclones_rep_pct2 <- cd8_topclones_rep_pct2 %>% 
  rownames_to_column("sample") %>% 
  mutate(sample = stringr::str_replace_all(sample, pattern = "Lymph nodes", replacement = "LN")) %>% 
  separate(sample, into = c("Cell_type","Organ","Strain","Exp")) %>% 
  mutate(Cell_type_organ = factor(paste(Organ, Strain), levels = c("Thymus WT","Thymus CA", "Thymus CAKR",
                                                                   "LN WT","LN CA", "LN CAKR" )))

cd8_topclones_rep_pct2

cd8_topclones_rep_pct2 %>% 
  filter(Organ == "LN") %>% 
  ggplot(aes(y = value*100, x = factor(Cell_type_organ))) +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  geom_point(aes(color = Strain)) + 
  ylab("% of repertoire") + 
  ggtitle("CD8 TRB") + 
  theme(axis.text.x = element_text(angle = 90)) + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) +
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,NA)) + xlab("")

# source data
write.csv(cd8_topclones_rep_pct2 %>% 
  filter(Organ == "LN") %>%
  mutate(value = value*100), 
  file = "topclones_cd8_trb.csv")

#ggsave("./final_fig/20topseqs/20topseqs_pct_trb_cd8.png", width = 5.4, height = 5.7, units = "cm")
#ggsave("./final_fig/20topseqs/20topseqs_pct_trb_cd8.eps", width = 5.4, height = 5.7, units = "cm")

```

# Heatmaps

In this part, we will use heatmaps to visualize the frequency of the sequences that were the most abundant (top20) in the periphery of WT mice in all samples. In the supplementary figure in the manuscript, reference samples (periphery of WT mice) are highlighted with black border, which was added manually in a graphics editor. 

## TRA

### CD4

Heatmap showing the frequency of the top20 CD4 TRA sequences from WT LNs in all samples. 

```{r fig.width=6, fig.height=8}

# TRA
cd4_topclones_heatmap_matrix <- prop.table.tra2 %>% 
  filter(aaSeqCDR3 %in% cd4_topclones_20sequences_tra) %>% 
  select(aaSeqCDR3, starts_with("CD4")) %>% 
  select(1,14,15,9:13,7,8,2:6)

cd4_topclones_heatmap_matrix2 <- as.matrix(cd4_topclones_heatmap_matrix[,2:15])
rownames(cd4_topclones_heatmap_matrix2) <- cd4_topclones_heatmap_matrix$aaSeqCDR3

cd4_topclones_heatmap_matrix2 <- cd4_topclones_heatmap_matrix2[match(cd4_topclones_20sequences_tra, rownames(cd4_topclones_heatmap_matrix2)),]

pheatmap(cd4_topclones_heatmap_matrix2, scale = "none", cluster_rows = F, cluster_cols = F)

#ggsave(plot = pheatmap(cd4_topclones_heatmap_matrix2, scale = "none", cluster_rows = F, cluster_cols = F), "final_fig/heatmap_top20_ordered/cd4_tra_noscale.svg")

```

### CD8 

Heatmap showing the frequency of the top20 CD8 TRA sequences from WT LNs in all samples. 

```{r fig.width=6, fig.height=8}
# TRA

cd8_topclones_heatmap_matrix <- prop.table.tra2 %>% 
  filter(aaSeqCDR3 %in% cd8_topclones_20sequences_tra) %>% 
  select(aaSeqCDR3, starts_with("CD8")) %>% 
  select(1,13:15,9:12,6:8,2:5)
  

cd8_topclones_heatmap_matrix2 <- as.matrix(cd8_topclones_heatmap_matrix[,2:15])
rownames(cd8_topclones_heatmap_matrix2) <- cd8_topclones_heatmap_matrix$aaSeqCDR3

cd8_topclones_heatmap_matrix2 <- cd8_topclones_heatmap_matrix2[match(cd8_topclones_20sequences_tra, rownames(cd8_topclones_heatmap_matrix2)),]

pheatmap(cd8_topclones_heatmap_matrix2, scale = "none", cluster_rows = F, cluster_cols = F)

#ggsave(plot = pheatmap(cd8_topclones_heatmap_matrix2, scale = "none", cluster_rows = F, cluster_cols = F), "final_fig/heatmap_top20_ordered/cd8_tra_noscale.svg")
```

As we can see that the first sequence takes up the most percentage of the repertoire, we will also plot another heatmap without the first sequence. 


```{r fig.width=6, fig.height=8}
### without 1st sequence

cd8_topclones_heatmap_matrix2 <- as.matrix(cd8_topclones_heatmap_matrix[2:20,2:15])
rownames(cd8_topclones_heatmap_matrix2) <- cd8_topclones_heatmap_matrix$aaSeqCDR3[2:20]

cd8_topclones_heatmap_matrix2 <- cd8_topclones_heatmap_matrix2[match(cd8_topclones_20sequences_tra[2:20], rownames(cd8_topclones_heatmap_matrix2)),]

pheatmap(cd8_topclones_heatmap_matrix2, scale = "none", cluster_rows = F, cluster_cols = F)

#ggsave(plot = pheatmap(cd8_topclones_heatmap_matrix2, scale = "none", cluster_rows = F, cluster_cols = F), "final_fig/heatmap_top20_ordered/cd8_tra_noscale_without.svg")
```

## TRB

### CD4

Heatmap showing the frequency of the top20 CD4 TRB sequences from WT LNs in all samples. 

```{r fig.width=6, fig.height=8}

# trb

cd4_topclones_heatmap_matrix <- prop.table.trb2 %>% filter(aaSeqCDR3 %in% cd4_topclones_20sequences_trb) %>% select(aaSeqCDR3, starts_with("CD4")) %>% select(1,14,15,9:13,7,8,2:6)
  
cd4_topclones_heatmap_matrix2 <- as.matrix(cd4_topclones_heatmap_matrix[,2:15])
rownames(cd4_topclones_heatmap_matrix2) <- cd4_topclones_heatmap_matrix$aaSeqCDR3

cd4_topclones_heatmap_matrix2 <- cd4_topclones_heatmap_matrix2[match( cd4_topclones_20sequences_trb, rownames(cd4_topclones_heatmap_matrix2)),]

pheatmap(cd4_topclones_heatmap_matrix2, scale = "none", cluster_rows = F, cluster_cols = F)

#ggsave(plot = pheatmap(cd4_topclones_heatmap_matrix2, scale = "none", cluster_rows = F, cluster_cols = F), "final_fig/heatmap_top20_ordered/cd4_trb_noscale.svg")
```

### CD8

Heatmap showing the frequency of the top20 CD8 TRB sequences from WT LNs in all samples. 

```{r fig.width=6, fig.height=8}
# trb

cd8_topclones_heatmap_matrix <- prop.table.trb2 %>% filter(aaSeqCDR3 %in% cd8_topclones_20sequences_trb) %>% select(aaSeqCDR3, starts_with("CD8")) %>% select(1,13:15,9:12,6:8,2:5)
  
cd8_topclones_heatmap_matrix2 <- as.matrix(cd8_topclones_heatmap_matrix[,2:15])
rownames(cd8_topclones_heatmap_matrix2) <- cd8_topclones_heatmap_matrix$aaSeqCDR3

cd8_topclones_heatmap_matrix2 <- cd8_topclones_heatmap_matrix2[match( cd8_topclones_20sequences_trb, rownames(cd8_topclones_heatmap_matrix2)),]

pheatmap(cd8_topclones_heatmap_matrix2, scale = "none", cluster_rows = F, cluster_cols = F)

# png("final_fig/heatmap_top20_ordered/cd8_trb_noscale.png", width = 400, height = 500)
# pheatmap(cd8_topclones_heatmap_matrix2, scale = "none", cluster_rows = F, cluster_cols = F)
# dev.off()

#ggsave(plot = pheatmap(cd8_topclones_heatmap_matrix2, scale = "none", cluster_rows = F, cluster_cols = F), "final_fig/heatmap_top20_ordered/cd8_trb_noscale.svg")
```

# Repertoire overlap

In this part we will investigate the overlap between repertoires of different samples. We will plot the percentage of overlapping TCR alpha or TCR beta CDR3 sequences in all pairs of samples in a matrix.

## TRA

Let's focus on the TRA now. We will remove iNKT sequences and keep only the information whether the CDR3 sequence is present in the sample or not (i.e., we will keep only unique occurrences).

```{r}
binary_tra <- excel_count_table_tra %>% 
  mutate(nkt_trav11_traj18 = if_else(
    (grepl(allVHitsWithScore, pattern = "TRAV11") | 
    (grepl(allJHitsWithScore, pattern = "TRAJ18"))),"yes","no")) %>% 
  filter(nkt_trav11_traj18 == "no") %>% 
  mutate_at(vars(starts_with("CD")), .funs = binary) 

binary_tra_longer <- binary_tra %>% select(starts_with("CD"), aaSeqCDR3) %>% pivot_longer(!aaSeqCDR3, names_to = "num_id")

```

We will loop through all the samples and calculate the percentage of overlapping sequences.

```{r}
df_overlap_tra <- data.frame("")

for(j in levels(factor(binary_tra_longer$num_id))){
  subset1 <- binary_tra_longer %>% filter(num_id == j & value>0) %>% pull("aaSeqCDR3")
  vector_overlap <- c()
  for(i in levels(factor(binary_tra_longer$num_id))){
    subset2 <- binary_tra_longer %>% filter(num_id == i & value>0) %>% pull("aaSeqCDR3")
    intersect_rep <- length(intersect(subset1, subset2))
    total <- length(subset2)
    vector_overlap <- c(vector_overlap,intersect_rep/total)
  }
  df <- as.data.frame(x = vector_overlap)
  colnames(df) <- j
  df
  df_overlap_tra <- cbind(df_overlap_tra, df)
}

# We have the matrix of overlap percentages, which we will use further for plotting
df_overlap_tra <- df_overlap_tra[,2:29]
rownames(df_overlap_tra) <- colnames(df_overlap_tra)

#write.csv(df_overlap_tra, "rep_overlap_tra.csv")
```

Now that we have the overlap matrix, we will plot the overlaps. First we will plot all the overlaps, i.e., the 28 samples that we have.

```{r fig.width=14, fig.height=12}

df_overlap_tra_for_plot <- df_overlap_tra

# Replace ones on the diagonal with zeroes
df_overlap_tra_for_plot[df_overlap_tra_for_plot == 1] <- 0

# Plot a dotplot of overlaps with percentages

df_overlap_tra_for_plot <- df_overlap_tra_for_plot %>% rownames_to_column("id") %>%  pivot_longer(-id)

g2 <- ggplot(df_overlap_tra_for_plot, 
             aes(id, 
                 factor(name, levels = rev(levels(factor(name)))))) + 
  geom_point(aes(size = value*100, colour = value*100)) + 
  theme_bw() 

g2 + scale_size_continuous(range=c(7,12)) +
  geom_text(aes(label = round(value*100, digits = 1))) + 
  scale_colour_gradient2(low = "lightskyblue", mid = "lightsteelblue2", high = "salmon") +
  theme(axis.text.x = element_text(angle = 90)) +
  xlab("") + ylab("")

```

### CD4 

As the previous matrix is too large and we are mostly interested in overlaps between CD4 samples and between CD8 samples, we will now perform a separated analysis.
 
```{r fig.width=8, fig.height=6}

rep_overlap_tra <- read_csv("rep_overlap_tra.csv") %>% column_to_rownames("...1")

rep_overlap_tra_cd4 <- rep_overlap_tra[1:14,1:14]
rep_overlap_tra_cd8 <- rep_overlap_tra[15:28,15:28]

rep_overlap_tra_cd4 <- rep_overlap_tra_cd4[ rev(c(5, 4, 3, 2, 1, 7, 6, 12, 11, 10, 9, 8, 14, 13)),
                                            rev(c(5, 4, 3, 2, 1, 7, 6, 12, 11, 10, 9, 8, 14, 13))]

levels_names <- colnames(rep_overlap_tra_cd4)

df_overlap_tra_cd4 <- rep_overlap_tra_cd4
df_overlap_tra_cd4[df_overlap_tra_cd4 == 1] <- 0

### plot dotplot of overlaps

df_overlap_tra_cd4 <- df_overlap_tra_cd4 %>% rownames_to_column("id") %>%  pivot_longer(-id)

g2 <- ggplot(df_overlap_tra_cd4, aes(x = factor(name, levels = levels_names), y = factor(id, levels = rev(levels_names)))) + 
  geom_point(aes(size = value*100, colour = value*100)) + 
  theme_bw() 

g2 + scale_size_continuous(range=c(5,9)) +
  geom_text(aes(label = round(value*100, digits = 1)), size = 3) + 
  scale_colour_gradient2(low = "white", mid = "lightsteelblue", high = "salmon") +
  theme(axis.text.x = element_text(angle = 90)) + ggtheme()

#ggsave(file = "./final_fig/overlap_cd4_tra.png", width = 20, height = 16, units = "cm")
#ggsave(file = "./final_fig/overlap_cd4_tra.svg", width = 20, height = 16, units = "cm")

```

### CD8 

```{r fig.width=8, fig.height=6}
rep_overlap_tra_cd8 <- rep_overlap_tra_cd8[rev(c(12:14,8:11,5:7,1:4)),rev(c(12:14,8:11,5:7,1:4))]

levels_names <- rev(colnames(rep_overlap_tra_cd8))

df_overlap_tra_cd8 <- rep_overlap_tra_cd8
df_overlap_tra_cd8[df_overlap_tra_cd8 == 1] <- 0

### plot dotplot of overlaps

df_overlap_tra_cd8 <- df_overlap_tra_cd8 %>% rownames_to_column("id") %>%  pivot_longer(-id)

g2 <- ggplot(df_overlap_tra_cd8, aes(factor(name, levels = levels_names), factor(id, levels = rev(levels_names)))) + 
  geom_point(aes(size = value*100, colour = value*100)) + 
  theme_bw() 

g2 + scale_size_continuous(range=c(5,9)) +
  geom_text(aes(label = round(value*100, digits = 1)), size = 3) + 
  scale_colour_gradient2(low = "white", mid = "lightsteelblue", high = "salmon") +
  theme(axis.text.x = element_text(angle = 90)) + ggtheme()

#ggsave(file = "./final_fig/overlap_cd8_tra.png", width = 20, height = 16, units = "cm")
#ggsave(file = "./final_fig/overlap_cd8_tra.svg", width = 20, height = 16, units = "cm")

```

### Overlap plots for the main figure

For the main figure we will create plots showing the percentage overlaps of WT samples with other samples. 

```{r fig.width=3, fig.height=3}
overlap_wt_tra_cd4 <- rep_overlap_tra_cd4 %>% 
  rownames_to_column("sample1") %>% 
  pivot_longer(!sample1, names_to = "sample2") %>% 
  mutate(sample2 = stringr::str_replace_all(sample2, pattern = "Lymph nodes", replacement = "LN"))  %>% 
  mutate(sample1 = stringr::str_replace_all(sample1, pattern = "Lymph nodes", replacement = "LN"))  %>% 
  separate(sample1, into = c("Cell_type1","Organ1","Strain1","Exp1"), remove = F)  %>% 
  separate(sample2, into = c("Cell_type2","Organ2","Strain2","Exp2"), remove = F) %>% 
  filter(value<1)

overlap_wt_tra_cd8 <- rep_overlap_tra_cd8 %>% 
  rownames_to_column("sample1") %>% 
  pivot_longer(!sample1, names_to = "sample2") %>% 
  mutate(sample2 = stringr::str_replace_all(sample2, pattern = "Lymph nodes", replacement = "LN"))  %>% 
  mutate(sample1 = stringr::str_replace_all(sample1, pattern = "Lymph nodes", replacement = "LN"))  %>% 
  separate(sample1, into = c("Cell_type1","Organ1","Strain1","Exp1"), remove = F)  %>% 
  separate(sample2, into = c("Cell_type2","Organ2","Strain2","Exp2"), remove = F) %>% 
  filter(value<1)

```

```{r fig.width=5, fig.height=4}
overlap_wt_tra_cd4 %>% filter(Organ1 == "Thymus" & Organ2 == "Thymus") %>% 
  mutate(organ_strain1 = paste(Organ1, Strain1), 
                          organ_strain2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strain1, organ_strain2, sep = " - ")) %>% 
  filter(comparison %in% c("Thymus WT - Thymus WT", "Thymus CA - Thymus WT", "Thymus CAKR - Thymus WT")) %>% 
  ggplot(aes(x = comparison, y = value)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=0) + 
  geom_jitter(position=position_jitter(0.2), size = 2, aes(color = comparison)) +
  ggtitle("CD4 TRA Thymus") +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  ylab("% repertoire overlap") + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,0.35)) + xlab("")


# source data
write.csv(overlap_wt_tra_cd4 %>% filter(Organ1 == "Thymus" & Organ2 == "Thymus") %>% 
  mutate(organ_strain1 = paste(Organ1, Strain1), 
                          organ_strain2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strain1, organ_strain2, sep = " - ")) %>% 
  filter(comparison %in% c("Thymus WT - Thymus WT", "Thymus CA - Thymus WT", "Thymus CAKR - Thymus WT")) %>% 
  select(sample1, sample2, comparison, value), 
  file = "source_data/rep_overlap_cd4_tra_thymus.csv",
  row.names = F)
  
#ggsave("./final_fig/overlaps/rep_overlap_cd4_tra_thy.png", width = 9.5, height = 9.5,  units = "cm")
#ggsave("./final_fig/overlaps/rep_overlap_cd4_tra_thy.eps", width = 9.5, height = 9.5,  units = "cm")

```


```{r fig.width=4, fig.height=3.5}
overlap_wt_tra_cd4 %>% filter(Organ1 == "LN" & Organ2 == "LN") %>% 
  mutate(organ_strain1 = paste(Organ1, Strain1), 
                          organ_strain2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strain1, organ_strain2, sep = " - ")) %>% 
  filter(comparison %in% c("LN WT - LN WT", "LN CA - LN WT", "LN CAKR - LN WT")) %>% 
  ggplot(aes(x = comparison, y = value)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=0) + 
  geom_jitter(position=position_jitter(0.2), size = 2, aes(color = comparison)) +
  ggtitle("CD4 TRA LN") +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  ylab("% repertoire overlap") + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,0.35)) + xlab("")

# source data
write.csv(overlap_wt_tra_cd4 %>% filter(Organ1 == "LN" & Organ2 == "LN") %>% 
  mutate(organ_strain1 = paste(Organ1, Strain1), 
                          organ_strain2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strain1, organ_strain2, sep = " - ")) %>% 
  filter(comparison %in% c("LN WT - LN WT", "LN CA - LN WT", "LN CAKR - LN WT")) %>% 
  select(sample1, sample2, comparison, value), 
  file = "source_data/rep_overlap_cd4_tra_ln.csv",
  row.names = F)

#ggsave("./final_fig/overlaps/rep_overlap_cd4_tra_LN.png", width = 8, height = 8,  units = "cm")
#ggsave("./final_fig/overlaps/rep_overlap_cd4_tra_LN.eps", width = 8, height = 8,  units = "cm")

```


```{r fig.width=5, fig.height=4}
overlap_wt_tra_cd8 %>% filter(Organ1 == "Thymus" & Organ2 == "Thymus") %>% 
  mutate(organ_strain1 = paste(Organ1, Strain1), 
                          organ_strain2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strain1, organ_strain2, sep = " - ")) %>% 
  filter(comparison %in% c("Thymus WT - Thymus WT", "Thymus CA - Thymus WT", "Thymus CAKR - Thymus WT")) %>% 
  ggplot(aes(x = comparison, y = value)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=0) + 
  geom_jitter(position=position_jitter(0.2), size = 2, aes(color = comparison)) +
  ggtitle("CD8 TRA Thymus") +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  ylab("% repertoire overlap") + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,0.35)) + xlab("")

# source data
write.csv(overlap_wt_tra_cd8 %>% filter(Organ1 == "Thymus" & Organ2 == "Thymus") %>% 
  mutate(organ_strain1 = paste(Organ1, Strain1), 
                          organ_strain2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strain1, organ_strain2, sep = " - ")) %>% 
  filter(comparison %in% c("Thymus WT - Thymus WT", "Thymus CA - Thymus WT", "Thymus CAKR - Thymus WT")) %>% 
  select(sample1, sample2, comparison, value), 
  file = "source_data/rep_overlap_cd8_tra_thymus.csv",
  row.names = F)

#ggsave("./final_fig/overlaps/rep_overlap_cd8_tra_thy.png", width = 9.5, height = 9.5,  units = "cm")
#ggsave("./final_fig/overlaps/rep_overlap_cd8_tra_thy.eps", width = 9.5, height = 9.5,  units = "cm")

```

```{r fig.width=4, fig.height=4}
overlap_wt_tra_cd8 %>% filter(Organ1 == "LN" & Organ2 == "LN") %>% 
  mutate(organ_strain1 = paste(Organ1, Strain1), 
                          organ_strain2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strain1, organ_strain2, sep = " - ")) %>% 
  filter(comparison %in% c("LN WT - LN WT", "LN CA - LN WT", "LN CAKR - LN WT")) %>% 
  ggplot(aes(x = comparison, y = value)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=0) + 
  geom_jitter(position=position_jitter(0.2), size = 2, aes(color = comparison)) +
  ggtitle("CD8 TRA LN") +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  ylab("% repertoire overlap") + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,0.35)) + xlab("")

# source data
write.csv(overlap_wt_tra_cd8 %>% filter(Organ1 == "LN" & Organ2 == "LN") %>% 
  mutate(organ_strain1 = paste(Organ1, Strain1), 
                          organ_strain2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strain1, organ_strain2, sep = " - ")) %>% 
  filter(comparison %in% c("LN WT - LN WT", "LN CA - LN WT", "LN CAKR - LN WT")) %>% 
  select(sample1, sample2, comparison, value), 
  file = "source_data/rep_overlap_cd8_tra_ln.csv",
  row.names = F)

#ggsave("./final_fig/overlaps/rep_overlap_cd8_tra_LN.png", width = 8, height = 8, units = "cm")
#ggsave("./final_fig/overlaps/rep_overlap_cd8_tra_LN.eps", width = 8, height = 8, units = "cm")

```

## TRB

Let's focus on the TRB now. The analysis follows the same logic as in case of TRA. No NKT sequence removal is applied in this case. We will keep only the information whether the CDR3 sequence is present in the sample or not (i.e., we will keep only unique occurrences). We will loop through all the samples and calculate the percentage of overlapping sequences.

```{r}

binary_trb <- excel_count_table_trb %>% 
  mutate_at(vars(starts_with("CD")), .funs = binary) 

binary_trb_longer <- binary_trb %>% select(starts_with("CD"), aaSeqCDR3) %>% pivot_longer(!aaSeqCDR3, names_to = "num_id")

df_overlap_trb <- data.frame("")

for(j in levels(factor(binary_trb_longer$num_id))){
  subset1 <- binary_trb_longer %>% filter(num_id == j & value>0) %>% pull("aaSeqCDR3")
  vector_overlap <- c()
  for(i in levels(factor(binary_trb_longer$num_id))){
    subset2 <- binary_trb_longer %>% filter(num_id == i & value>0) %>% pull("aaSeqCDR3")
    intersect_rep <- length(intersect(subset1, subset2))
    total <- length(subset2)
    vector_overlap <- c(vector_overlap,intersect_rep/total)
  }
  df <- as.data.frame(x = vector_overlap)
  colnames(df) <- j
  df
  df_overlap_trb <- cbind(df_overlap_trb, df)
}

df_overlap_trb <- df_overlap_trb[,2:29]
rownames(df_overlap_trb) <- colnames(df_overlap_trb)

#write.csv(df_overlap_trb, "rep_overlap_trb.csv")
```


```{r fig.width=14, fig.height=12}
df_overlap_trb_for_plot <- df_overlap_trb
df_overlap_trb_for_plot[df_overlap_trb_for_plot == 1] <- 0

### plot dotplot of overlaps

df_overlap_trb_for_plot <- df_overlap_trb_for_plot %>% rownames_to_column("id") %>%  pivot_longer(-id)

g2 <- ggplot(df_overlap_trb_for_plot, aes(id, factor(name, levels = rev(levels(factor(name)))))) + 
  geom_point(aes(size = value*100, colour = value*100)) + 
  theme_bw() 

g2 + scale_size_continuous(range=c(7,12)) +
  geom_text(aes(label = round(value*100, digits = 1))) + 
  scale_colour_gradient2(low = "lightskyblue", mid = "lightsteelblue2", high = "salmon") +
  theme(axis.text.x = element_text(angle = 90)) 


```

### CD4

```{r fig.width=8, fig.height=6}

rep_overlap_trb <- read_csv("rep_overlap_trb.csv") %>% column_to_rownames("...1")

rep_overlap_trb_cd4 <- rep_overlap_trb[1:14,1:14]
rep_overlap_trb_cd8 <- rep_overlap_trb[15:28,15:28]

rep_overlap_trb_cd4 <- rep_overlap_trb_cd4[rev(c(13,14,8:12,6,7,1:5)),rev(c(13,14,8:12,6,7,1:5))]

levels_names <- rev(colnames(rep_overlap_trb_cd4))


df_overlap_trb_cd4 <- rep_overlap_trb_cd4
df_overlap_trb_cd4[df_overlap_trb_cd4 == 1] <- 0

### plot dotplot of overlaps

df_overlap_trb_cd4 <- df_overlap_trb_cd4 %>% rownames_to_column("id") %>%  pivot_longer(-id)

g2 <- ggplot(df_overlap_trb_cd4, aes(factor(name, levels = levels_names), factor(id, levels = rev(levels_names)))) + 
  geom_point(aes(size = value*100, colour = value*100)) + 
  theme_bw() 

g2 + scale_size_continuous(range=c(5,9)) +
  geom_text(aes(label = round(value*100, digits = 1)), size = 3) + 
  scale_colour_gradient2(low = "white", mid = "lightsteelblue", high = "salmon") +
  theme(axis.text.x = element_text(angle = 90)) + ggtheme()

#ggsave(file = "./final_fig/overlap_cd4_trb.png", width = 20, height = 16, units = "cm")
#ggsave(file = "./final_fig/overlap_cd4_trb.svg", width = 20, height = 16, units = "cm")

```

### CD8

```{r fig.width=8, fig.height=6}
rep_overlap_trb_cd8 <- rep_overlap_trb_cd8[rev(c(12:14,8:11,5:7,1:4)),rev(c(12:14,8:11,5:7,1:4))]

levels_names <- rev(colnames(rep_overlap_trb_cd8))

df_overlap_trb_cd8 <- rep_overlap_trb_cd8
df_overlap_trb_cd8[df_overlap_trb_cd8 == 1] <- 0

### plot dotplot of overlaps

df_overlap_trb_cd8 <- df_overlap_trb_cd8 %>% rownames_to_column("id") %>%  pivot_longer(-id)

g2 <- ggplot(df_overlap_trb_cd8, aes(factor(name, levels = levels_names), factor(id, levels = rev(levels_names)))) + 
  geom_point(aes(size = value*100, colour = value*100)) + 
  theme_bw() 

g2 + scale_size_continuous(range=c(5,9)) +
  geom_text(aes(label = round(value*100, digits = 1)), size = 3) + 
  scale_colour_gradient2(low = "white", mid = "lightsteelblue", high = "salmon") +
  theme(axis.text.x = element_text(angle = 90)) + ggtheme()

#ggsave(file = "./final_fig/overlap_cd8_trb.png", width = 20, height = 16, units = "cm")
#ggsave(file = "./final_fig/overlap_cd8_trb.svg", width = 20, height = 16, units = "cm")

```

### Overlap plots for the main figure

For the main figure we will create plots showing the percentage overlaps of WT samples with other samples. 

```{r}

overlap_wt_trb_cd4 <- rep_overlap_trb_cd4 %>% 
  rownames_to_column("sample1") %>% 
  pivot_longer(!sample1, names_to = "sample2") %>% 
  mutate(sample2 = stringr::str_replace_all(sample2, pattern = "Lymph nodes", replacement = "LN"))  %>% 
  mutate(sample1 = stringr::str_replace_all(sample1, pattern = "Lymph nodes", replacement = "LN"))  %>% 
  separate(sample1, into = c("Cell_type1","Organ1","Strain1","Exp1"), remove = F)  %>% 
  separate(sample2, into = c("Cell_type2","Organ2","Strain2","Exp2"), remove = F) %>% 
  filter(value<1)

overlap_wt_trb_cd8 <- rep_overlap_trb_cd8 %>% 
  rownames_to_column("sample1") %>% 
  pivot_longer(!sample1, names_to = "sample2") %>% 
  mutate(sample2 = stringr::str_replace_all(sample2, pattern = "Lymph nodes", replacement = "LN"))  %>% 
  mutate(sample1 = stringr::str_replace_all(sample1, pattern = "Lymph nodes", replacement = "LN"))  %>% 
  separate(sample1, into = c("Cell_type1","Organ1","Strain1","Exp1"), remove = F)  %>% 
  separate(sample2, into = c("Cell_type2","Organ2","Strain2","Exp2"), remove = F) %>% 
  filter(value<1)
```

```{r fig.width=5, fig.height=4}
overlap_wt_trb_cd4 %>% filter(Organ1 == "Thymus" & Organ2 == "Thymus") %>% 
  mutate(organ_strbin1 = paste(Organ1, Strain1), 
                          organ_strbin2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strbin1, organ_strbin2, sep = " - ")) %>% 
  filter(comparison %in% c("Thymus WT - Thymus WT", "Thymus CA - Thymus WT", "Thymus CAKR - Thymus WT")) %>% 
  ggplot(aes(x = comparison, y = value)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=0) + 
  geom_jitter(position=position_jitter(0.2), size = 2, aes(color = comparison)) +
  ggtitle("CD4 TRB Thymus") +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  ylab("% repertoire overlap") + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,0.20)) + xlab("")

# source data
write.csv(overlap_wt_trb_cd4 %>% filter(Organ1 == "Thymus" & Organ2 == "Thymus") %>% 
  mutate(organ_strbin1 = paste(Organ1, Strain1), 
                          organ_strbin2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strbin1, organ_strbin2, sep = " - ")) %>% 
  filter(comparison %in% c("Thymus WT - Thymus WT", "Thymus CA - Thymus WT", "Thymus CAKR - Thymus WT")) %>% 
  select(sample1, sample2, comparison, value), 
  file = "source_data/rep_overlap_cd4_trb_thymus.csv",
  row.names = F)

#ggsave("./final_fig/overlaps/rep_overlap_cd4_trb_thy.png", width = 9.5, height = 9.5,  units = "cm")
#ggsave("./final_fig/overlaps/rep_overlap_cd4_trb_thy.eps", width = 9.5, height = 9.5,  units = "cm")

```

```{r fig.width=4, fig.height=3.5}
overlap_wt_trb_cd4 %>% filter(Organ1 == "LN" & Organ2 == "LN") %>% 
  mutate(organ_strbin1 = paste(Organ1, Strain1), 
                          organ_strbin2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strbin1, organ_strbin2, sep = " - ")) %>% 
  filter(comparison %in% c("LN WT - LN WT", "LN CA - LN WT", "LN CAKR - LN WT")) %>% 
  ggplot(aes(x = comparison, y = value)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=0) + 
  geom_jitter(position=position_jitter(0.2), size = 2, aes(color = comparison)) +
  ggtitle("CD4 TRB LN") +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  ylab("% repertoire overlap") + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,0.20)) + xlab("")

# source data
write.csv(overlap_wt_trb_cd4 %>% filter(Organ1 == "LN" & Organ2 == "LN") %>% 
  mutate(organ_strbin1 = paste(Organ1, Strain1), 
                          organ_strbin2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strbin1, organ_strbin2, sep = " - ")) %>% 
  filter(comparison %in% c("LN WT - LN WT", "LN CA - LN WT", "LN CAKR - LN WT")) %>% 
  select(sample1, sample2, comparison, value), 
  file = "source_data/rep_overlap_cd4_trb_ln.csv",
  row.names = F)

#ggsave("./final_fig/overlaps/rep_overlap_cd4_trb_LN.png", width = 8, height = 8,  units = "cm")
#ggsave("./final_fig/overlaps/rep_overlap_cd4_trb_LN.eps", width = 8, height = 8,  units = "cm")

```


```{r fig.width=5, fig.height=4}
overlap_wt_trb_cd8 %>% filter(Organ1 == "Thymus" & Organ2 == "Thymus") %>% 
  mutate(organ_strbin1 = paste(Organ1, Strain1), 
                          organ_strbin2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strbin1, organ_strbin2, sep = " - ")) %>% 
  filter(comparison %in% c("Thymus WT - Thymus WT", "Thymus CA - Thymus WT", "Thymus CAKR - Thymus WT")) %>% 
  ggplot(aes(x = comparison, y = value)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=0) + 
  geom_jitter(position=position_jitter(0.2), size = 2, aes(color = comparison)) +
  ggtitle("CD8 TRB Thymus") +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  ylab("% repertoire overlap") + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,0.25)) + xlab("")

# source data
write.csv(overlap_wt_trb_cd8 %>% filter(Organ1 == "Thymus" & Organ2 == "Thymus") %>% 
  mutate(organ_strbin1 = paste(Organ1, Strain1), 
                          organ_strbin2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strbin1, organ_strbin2, sep = " - ")) %>% 
  filter(comparison %in% c("Thymus WT - Thymus WT", "Thymus CA - Thymus WT", "Thymus CAKR - Thymus WT")) %>% 
  select(sample1, sample2, comparison, value), 
  file = "source_data/rep_overlap_cd8_trb_thymus.csv",
  row.names = F)

#ggsave("./final_fig/overlaps/rep_overlap_cd8_trb_thy.png", width = 9.5, height = 9.5,  units = "cm")
#ggsave("./final_fig/overlaps/rep_overlap_cd8_trb_thy.eps", width = 9.5, height = 9.5,  units = "cm")

```

```{r fig.width=4, fig.height=3.5}
overlap_wt_trb_cd8 %>% filter(Organ1 == "LN" & Organ2 == "LN") %>% 
  mutate(organ_strbin1 = paste(Organ1, Strain1), 
                          organ_strbin2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strbin1, organ_strbin2, sep = " - ")) %>% 
  filter(comparison %in% c("LN WT - LN WT", "LN CA - LN WT", "LN CAKR - LN WT")) %>% 
  ggplot(aes(x = comparison, y = value)) +
  geom_dotplot(binaxis='y', stackdir='center', dotsize=0) + 
  geom_jitter(position=position_jitter(0.2), size = 2, aes(color = comparison)) +
  ggtitle("CD8 TRB LN") +
  stat_summary(fun = "mean",
               geom = "crossbar", 
               width = 0.5,
               colour = "black") +
  ylab("% repertoire overlap") + 
  theme_classic() + 
  ggtheme() + 
  theme(axis.text.x = element_text(angle = 90)) + 
  scale_color_manual(values = c("dodgerblue","indianred2","gray40")) + 
  ylim(c(0,0.25)) + xlab("")

# source data
write.csv(overlap_wt_trb_cd8 %>% filter(Organ1 == "LN" & Organ2 == "LN") %>% 
  mutate(organ_strbin1 = paste(Organ1, Strain1), 
                          organ_strbin2 = paste(Organ2, Strain2)) %>% 
  mutate(comparison = paste(organ_strbin1, organ_strbin2, sep = " - ")) %>% 
  filter(comparison %in% c("LN WT - LN WT", "LN CA - LN WT", "LN CAKR - LN WT")) %>% 
  select(sample1, sample2, comparison, value), 
  file = "source_data/rep_overlap_cd8_trb_ln.csv",
  row.names = F)

#ggsave("./final_fig/overlaps/rep_overlap_cd8_trb_LN.png", width = 8, height = 8,  units = "cm")
#ggsave("./final_fig/overlaps/rep_overlap_cd8_trb_LN.eps", width = 8, height = 8,  units = "cm")

```


# SessionInfo

```{r}
sessionInfo()
```

